---
knit: bookdown::preview_chapter
---

# Expression overview

## Introduction

Here we will continue to work with the filtered __blischak__ dataset produced in the previous chapter. We will look at what happened to the expression matrix after the quality control step. Ideally, we expect to see batches from the same individual grouping together and distinct groups corresponding to each individual. 

```{r, echo=FALSE}
library(knitr)
opts_chunk$set(out.width='90%', fig.align = 'center')
```

```{r, message=FALSE, warning=FALSE}
library(scater, quietly = TRUE)
options(stringsAsFactors = FALSE)
umi <- readRDS("blischak/umi.rds")
umi.qc <- umi[fData(umi)$use, pData(umi)$use]
endog_genes <- !fData(umi.qc)$is_feature_control
```

## PCA plot

The easiest thing to overview the data is to transform it using the principal component analysis and then visualize the first two principal components. Again the [scater](https://github.com/davismcc/scater) package provides several very useful functions to make this analysis straightforward. 

### Before QC

```{r expr-overview-pca-before-qc, fig.cap = "PCA plot of the blischak data"}
scater::plotPCA(umi[endog_genes, ],
                ntop = 500,
                colour_by = "batch",
                size_by = "total_features",
                shape_by = "individual",
                exprs_values = "counts")
```

### After QC

```{r expr-overview-pca-after-qc, fig.cap = "PCA plot of the blischak data"}
scater::plotPCA(umi.qc[endog_genes, ],
                ntop = 500,
                colour_by = "batch",
                size_by = "total_features",
                shape_by = "individual",
                exprs_values = "counts")
```

By default only the top 500 most variable genes are used to calculate the PCA. This can be adjusted by changing the `ntop` argument. 

__Exercise 1__
How do the PCA plots change if when all 14,214 genes are used? Or when only top 50 genes are used?

__Our answer__

```{r expr-overview-pca-after-qc-exercise1-1, fig.cap = "PCA plot of the blischak data (14214 genes)", echo=FALSE}
scater::plotPCA(umi.qc[endog_genes, ],
                ntop = 14214,
                colour_by = "batch",
                size_by = "total_features",
                shape_by = "individual",
                exprs_values = "counts")
```

```{r expr-overview-pca-after-qc-exercise1-2, fig.cap = "PCA plot of the blischak data (50 genes)", echo=FALSE}
scater::plotPCA(umi.qc[endog_genes, ],
                ntop = 50,
                colour_by = "batch",
                size_by = "total_features",
                shape_by = "individual",
                exprs_values = "counts")
```

If your answers are different please compare your code with [ours](https://github.com/hemberg-lab/scRNA.seq.course/blob/master/07-exprs-overview.Rmd) (you need to search for this exercise in the opened file).

## tSNE map

An alternative to PCA for visualizing scRNASeq data is a tSNE plot. [tSNE](https://lvdmaaten.github.io/tsne/) (t-Distributed Stochastic Neighbor Embedding) combines dimensionality reduction (e.g. PCA) with random walks on the nearest-neighbour network to map high dimensional data (i.e. our 14,214 dimensional expression matrix) to a 2-dimensional space while preserving local distances between cells. In contrast with PCA, tSNE is a stochastic algorithm which means running the method multiple times on the same dataset will result in different plots. In the code below we fix the "seed" of the random-number generator so that we always get the same plot. 



### Before QC

```{r expr-overview-tsne-before-qc, fig.cap = "tSNE map of the blischak data"}
scater::plotTSNE(umi[endog_genes, ],
                 ntop = 500,
                 perplexity = 130,
                 colour_by = "batch",
                 size_by = "total_features",
                 shape_by = "individual",
                 exprs_values = "counts",
                 rand_seed = 123456)
```

### After QC

```{r expr-overview-tsne-after-qc, fig.cap = "tSNE map of the blischak data"}
scater::plotTSNE(umi.qc[endog_genes, ],
                 ntop = 500,
                 perplexity = 130,
                 colour_by = "batch",
                 size_by = "total_features",
                 shape_by = "individual",
                 exprs_values = "counts",
                 rand_seed = 123456)
```

Furthermore tSNE requires you to provide a value of "perplexity" which reflects the number of neighbours used to build the nearest-neighbour network; a high value creates a dense network which clumps cells together while a low value makes the network more sparse allowing groups of cells to separate from each other. __scater__ uses a default perplexity of the total number of cells divided by five (rounded down).

__Exercise 2__
How do the tSNE plots change when a perplexity of 10 or 500 is used?

__Our answer__

```{r expr-overview-tsne-after-qc-exercise2-1, fig.cap = "tSNE map of the blischak data (perplexity = 10)", echo=FALSE}
scater::plotTSNE(umi.qc[endog_genes, ],
                 ntop = 500,
                 perplexity = 10,
                 colour_by = "batch",
                 size_by = "total_features",
                 shape_by = "individual",
                 exprs_values = "counts",
                 rand_seed = 123456)
```

```{r expr-overview-tsne-after-qc-exercise2-2, fig.cap = "tSNE map of the blischak data (perplexity = 500)", echo=FALSE}
scater::plotTSNE(umi.qc[endog_genes, ],
                 ntop = 500,
                 perplexity = 400,
                 colour_by = "batch",
                 size_by = "total_features",
                 shape_by = "individual",
                 exprs_values = "counts",
                 rand_seed = 123456)
```

If your answers are different please compare your code with [ours](https://github.com/hemberg-lab/scRNA.seq.course/blob/master/07-exprs-overview.Rmd) (you need to search for this exercise in the opened file).

## Big Exercise

Perform the same analysis with read counts of the Blischak data. Use `blischak/reads.rds` file to load the reads SCESet object. Once you have finished please compare your results to ours (next chapter).
