<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Analysis of single-cell RNA-seq data</title>
  <meta content="text/html; charset=UTF-8" http-equiv="Content-Type">
  <meta name="description" content="Analysis of single-cell RNA-seq data">
  <meta name="generator" content="bookdown 0.0.53 and GitBook 2.6.7">

<meta name="author" content="Vladimir Kiselev, Tallulah Andrews and Martin Hemberg">

<meta name="date" content="2016-03-23">

<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  TeX: { equationNumbers: { autoNumber: "AMS" } }
});
</script>
  <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"></script>

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="confounding-factors-reads.html">
<link rel="next" href="normalization-for-library-size-reads.html">

<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>


  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="index.html">Table of Contents</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About the course</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#registration"><i class="fa fa-check"></i><b>1.1</b> Registration</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#github"><i class="fa fa-check"></i><b>1.2</b> GitHub</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>1.3</b> License</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#prerequisites"><i class="fa fa-check"></i><b>1.4</b> Prerequisites</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="technical-requirements.html"><a href="technical-requirements.html"><i class="fa fa-check"></i><b>2</b> Technical requirements</a></li>
<li class="chapter" data-level="3" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html"><i class="fa fa-check"></i><b>3</b> Introduction to single-cell RNA-seq</a><ul>
<li class="chapter" data-level="3.1" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#bulk-rna-seq"><i class="fa fa-check"></i><b>3.1</b> Bulk RNA-seq</a></li>
<li class="chapter" data-level="3.2" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#scrna-seq"><i class="fa fa-check"></i><b>3.2</b> scRNA-seq</a></li>
<li class="chapter" data-level="3.3" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#protocol"><i class="fa fa-check"></i><b>3.3</b> Protocol</a></li>
<li class="chapter" data-level="3.4" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#analysis"><i class="fa fa-check"></i><b>3.4</b> Analysis</a></li>
<li class="chapter" data-level="3.5" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#challenges"><i class="fa fa-check"></i><b>3.5</b> Challenges</a></li>
<li class="chapter" data-level="3.6" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#controls"><i class="fa fa-check"></i><b>3.6</b> Controls</a><ul>
<li class="chapter" data-level="3.6.1" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#ercc-spike-ins"><i class="fa fa-check"></i><b>3.6.1</b> ERCC <em>spike-ins</em></a></li>
<li class="chapter" data-level="3.6.2" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#umis"><i class="fa fa-check"></i><b>3.6.2</b> UMIs</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html"><i class="fa fa-check"></i><b>4</b> Construction of expression matrix</a><ul>
<li class="chapter" data-level="4.1" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#reads-qc"><i class="fa fa-check"></i><b>4.1</b> Reads QC</a></li>
<li class="chapter" data-level="4.2" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#reads-alignment"><i class="fa fa-check"></i><b>4.2</b> Reads alignment</a></li>
<li class="chapter" data-level="4.3" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#alignment-example"><i class="fa fa-check"></i><b>4.3</b> Alignment example</a></li>
<li class="chapter" data-level="4.4" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#mapping-qc"><i class="fa fa-check"></i><b>4.4</b> Mapping QC</a></li>
<li class="chapter" data-level="4.5" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#reads-quantification"><i class="fa fa-check"></i><b>4.5</b> Reads quantification</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="scater-package.html"><a href="scater-package.html"><i class="fa fa-check"></i><b>5</b> scater package</a><ul>
<li class="chapter" data-level="5.1" data-path="scater-package.html"><a href="scater-package.html#introduction"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="scater-package.html"><a href="scater-package.html#scater-workflow"><i class="fa fa-check"></i><b>5.2</b> scater workflow</a></li>
<li class="chapter" data-level="5.3" data-path="scater-package.html"><a href="scater-package.html#terminology"><i class="fa fa-check"></i><b>5.3</b> Terminology</a></li>
<li class="chapter" data-level="5.4" data-path="scater-package.html"><a href="scater-package.html#sceset-class"><i class="fa fa-check"></i><b>5.4</b> SCESet class</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html"><i class="fa fa-check"></i><b>6</b> Expression QC (UMI)</a><ul>
<li class="chapter" data-level="6.1" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#introduction-1"><i class="fa fa-check"></i><b>6.1</b> Introduction</a></li>
<li class="chapter" data-level="6.2" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#blischak-dataset"><i class="fa fa-check"></i><b>6.2</b> Blischak dataset</a></li>
<li class="chapter" data-level="6.3" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#gene-qc"><i class="fa fa-check"></i><b>6.3</b> Gene QC</a><ul>
<li class="chapter" data-level="6.3.1" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#gene-dropouts"><i class="fa fa-check"></i><b>6.3.1</b> Gene dropouts</a></li>
<li class="chapter" data-level="6.3.2" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#gene-expression"><i class="fa fa-check"></i><b>6.3.2</b> Gene expression</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#gene-filtering"><i class="fa fa-check"></i><b>6.4</b> Gene filtering</a></li>
<li class="chapter" data-level="6.5" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#cell-qc"><i class="fa fa-check"></i><b>6.5</b> Cell QC</a><ul>
<li class="chapter" data-level="6.5.1" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#library-size"><i class="fa fa-check"></i><b>6.5.1</b> Library size</a></li>
<li class="chapter" data-level="6.5.2" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#detected-genes-1"><i class="fa fa-check"></i><b>6.5.2</b> Detected genes (1)</a></li>
<li class="chapter" data-level="6.5.3" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#detected-genes-2"><i class="fa fa-check"></i><b>6.5.3</b> Detected genes (2)</a></li>
<li class="chapter" data-level="6.5.4" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#erccs-and-mts"><i class="fa fa-check"></i><b>6.5.4</b> ERCCs and MTs</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#cell-filtering"><i class="fa fa-check"></i><b>6.6</b> Cell filtering</a><ul>
<li class="chapter" data-level="6.6.1" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#default"><i class="fa fa-check"></i><b>6.6.1</b> Default</a></li>
<li class="chapter" data-level="6.6.2" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#automatic"><i class="fa fa-check"></i><b>6.6.2</b> Automatic</a></li>
<li class="chapter" data-level="6.6.3" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#manual"><i class="fa fa-check"></i><b>6.6.3</b> Manual</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#compare-filterings"><i class="fa fa-check"></i><b>6.7</b> Compare filterings</a></li>
<li class="chapter" data-level="6.8" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#save-the-data"><i class="fa fa-check"></i><b>6.8</b> Save the data</a></li>
<li class="chapter" data-level="6.9" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#exercise"><i class="fa fa-check"></i><b>6.9</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="expression-qc-reads.html"><a href="expression-qc-reads.html"><i class="fa fa-check"></i><b>7</b> Expression QC (Reads)</a></li>
<li class="chapter" data-level="8" data-path="expression-overview.html"><a href="expression-overview.html"><i class="fa fa-check"></i><b>8</b> Expression overview</a><ul>
<li class="chapter" data-level="8.1" data-path="expression-overview.html"><a href="expression-overview.html#introduction-2"><i class="fa fa-check"></i><b>8.1</b> Introduction</a></li>
<li class="chapter" data-level="8.2" data-path="expression-overview.html"><a href="expression-overview.html#top-500-genes"><i class="fa fa-check"></i><b>8.2</b> Top 500 genes</a><ul>
<li class="chapter" data-level="8.2.1" data-path="expression-overview.html"><a href="expression-overview.html#before-qc"><i class="fa fa-check"></i><b>8.2.1</b> Before QC</a></li>
<li class="chapter" data-level="8.2.2" data-path="expression-overview.html"><a href="expression-overview.html#after-qc"><i class="fa fa-check"></i><b>8.2.2</b> After QC</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="expression-overview.html"><a href="expression-overview.html#pca-plot"><i class="fa fa-check"></i><b>8.3</b> PCA plot</a><ul>
<li class="chapter" data-level="8.3.1" data-path="expression-overview.html"><a href="expression-overview.html#before-qc-1"><i class="fa fa-check"></i><b>8.3.1</b> Before QC</a></li>
<li class="chapter" data-level="8.3.2" data-path="expression-overview.html"><a href="expression-overview.html#after-qc-1"><i class="fa fa-check"></i><b>8.3.2</b> After QC</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="expression-overview.html"><a href="expression-overview.html#diffusion-map"><i class="fa fa-check"></i><b>8.4</b> Diffusion map</a><ul>
<li class="chapter" data-level="8.4.1" data-path="expression-overview.html"><a href="expression-overview.html#before-qc-2"><i class="fa fa-check"></i><b>8.4.1</b> Before QC</a></li>
<li class="chapter" data-level="8.4.2" data-path="expression-overview.html"><a href="expression-overview.html#after-qc-2"><i class="fa fa-check"></i><b>8.4.2</b> After QC</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="expression-overview.html"><a href="expression-overview.html#tsne-map"><i class="fa fa-check"></i><b>8.5</b> tSNE map</a><ul>
<li class="chapter" data-level="8.5.1" data-path="expression-overview.html"><a href="expression-overview.html#before-qc-3"><i class="fa fa-check"></i><b>8.5.1</b> Before QC</a></li>
<li class="chapter" data-level="8.5.2" data-path="expression-overview.html"><a href="expression-overview.html#after-qc-3"><i class="fa fa-check"></i><b>8.5.2</b> After QC</a></li>
</ul></li>
<li class="chapter" data-level="8.6" data-path="expression-overview.html"><a href="expression-overview.html#exercise-1"><i class="fa fa-check"></i><b>8.6</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="expression-overview-reads.html"><a href="expression-overview-reads.html"><i class="fa fa-check"></i><b>9</b> Expression overview (Reads)</a></li>
<li class="chapter" data-level="10" data-path="confounding-factors.html"><a href="confounding-factors.html"><i class="fa fa-check"></i><b>10</b> Confounding factors</a><ul>
<li class="chapter" data-level="10.1" data-path="confounding-factors.html"><a href="confounding-factors.html#introduction-3"><i class="fa fa-check"></i><b>10.1</b> Introduction</a></li>
<li class="chapter" data-level="10.2" data-path="confounding-factors.html"><a href="confounding-factors.html#correlations-with-pcs"><i class="fa fa-check"></i><b>10.2</b> Correlations with PCs</a><ul>
<li class="chapter" data-level="10.2.1" data-path="confounding-factors.html"><a href="confounding-factors.html#detected-genes"><i class="fa fa-check"></i><b>10.2.1</b> Detected genes</a></li>
<li class="chapter" data-level="10.2.2" data-path="confounding-factors.html"><a href="confounding-factors.html#control-erccs"><i class="fa fa-check"></i><b>10.2.2</b> Control ERCCs</a></li>
<li class="chapter" data-level="10.2.3" data-path="confounding-factors.html"><a href="confounding-factors.html#control-mts"><i class="fa fa-check"></i><b>10.2.3</b> Control MTs</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="confounding-factors.html"><a href="confounding-factors.html#explanatory-variables"><i class="fa fa-check"></i><b>10.3</b> Explanatory variables</a></li>
<li class="chapter" data-level="10.4" data-path="confounding-factors.html"><a href="confounding-factors.html#other-confounders"><i class="fa fa-check"></i><b>10.4</b> Other confounders</a></li>
<li class="chapter" data-level="10.5" data-path="confounding-factors.html"><a href="confounding-factors.html#exercise-2"><i class="fa fa-check"></i><b>10.5</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="confounding-factors-reads.html"><a href="confounding-factors-reads.html"><i class="fa fa-check"></i><b>11</b> Confounding factors (Reads)</a></li>
<li class="chapter" data-level="12" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html"><i class="fa fa-check"></i><b>12</b> Normalization for library size</a><ul>
<li class="chapter" data-level="12.1" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#introduction-4"><i class="fa fa-check"></i><b>12.1</b> Introduction</a></li>
<li class="chapter" data-level="12.2" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#library-size-1"><i class="fa fa-check"></i><b>12.2</b> Library size</a></li>
<li class="chapter" data-level="12.3" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#normalisations"><i class="fa fa-check"></i><b>12.3</b> Normalisations</a><ul>
<li class="chapter" data-level="12.3.1" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#raw"><i class="fa fa-check"></i><b>12.3.1</b> Raw</a></li>
<li class="chapter" data-level="12.3.2" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#cpm"><i class="fa fa-check"></i><b>12.3.2</b> CPM</a></li>
<li class="chapter" data-level="12.3.3" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#log2cpm"><i class="fa fa-check"></i><b>12.3.3</b> log2(CPM)</a></li>
<li class="chapter" data-level="12.3.4" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#tmm"><i class="fa fa-check"></i><b>12.3.4</b> TMM</a></li>
<li class="chapter" data-level="12.3.5" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#rle"><i class="fa fa-check"></i><b>12.3.5</b> RLE</a></li>
<li class="chapter" data-level="12.3.6" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#upperquantile"><i class="fa fa-check"></i><b>12.3.6</b> Upperquantile</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#interpretation"><i class="fa fa-check"></i><b>12.4</b> Interpretation</a></li>
<li class="chapter" data-level="12.5" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#other-methods"><i class="fa fa-check"></i><b>12.5</b> Other methods</a></li>
<li class="chapter" data-level="12.6" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#visualize-genes"><i class="fa fa-check"></i><b>12.6</b> Visualize genes</a><ul>
<li class="chapter" data-level="12.6.1" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#raw-1"><i class="fa fa-check"></i><b>12.6.1</b> Raw</a></li>
<li class="chapter" data-level="12.6.2" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#cpm-1"><i class="fa fa-check"></i><b>12.6.2</b> CPM</a></li>
<li class="chapter" data-level="12.6.3" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#log2cpm-1"><i class="fa fa-check"></i><b>12.6.3</b> log2(CPM)</a></li>
<li class="chapter" data-level="12.6.4" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#upperquantile-1"><i class="fa fa-check"></i><b>12.6.4</b> Upperquantile</a></li>
<li class="chapter" data-level="12.6.5" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#fpkm"><i class="fa fa-check"></i><b>12.6.5</b> FPKM</a></li>
<li class="chapter" data-level="12.6.6" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#tpm"><i class="fa fa-check"></i><b>12.6.6</b> TPM</a></li>
</ul></li>
<li class="chapter" data-level="12.7" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#exercise-3"><i class="fa fa-check"></i><b>12.7</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="normalization-for-library-size-reads.html"><a href="normalization-for-library-size-reads.html"><i class="fa fa-check"></i><b>13</b> Normalization for library size (Reads)</a></li>
<li class="chapter" data-level="14" data-path="remove-confounders-using-controls.html"><a href="remove-confounders-using-controls.html"><i class="fa fa-check"></i><b>14</b> Remove confounders using controls</a><ul>
<li class="chapter" data-level="14.1" data-path="remove-confounders-using-controls.html"><a href="remove-confounders-using-controls.html#introduction-5"><i class="fa fa-check"></i><b>14.1</b> Introduction</a></li>
<li class="chapter" data-level="14.2" data-path="remove-confounders-using-controls.html"><a href="remove-confounders-using-controls.html#brennecke-method"><i class="fa fa-check"></i><b>14.2</b> Brennecke method</a></li>
<li class="chapter" data-level="14.3" data-path="remove-confounders-using-controls.html"><a href="remove-confounders-using-controls.html#remove-unwanted-variation"><i class="fa fa-check"></i><b>14.3</b> Remove Unwanted Variation</a></li>
<li class="chapter" data-level="14.4" data-path="remove-confounders-using-controls.html"><a href="remove-confounders-using-controls.html#effectiveness-1"><i class="fa fa-check"></i><b>14.4</b> Effectiveness 1</a></li>
<li class="chapter" data-level="14.5" data-path="remove-confounders-using-controls.html"><a href="remove-confounders-using-controls.html#effectiveness-2"><i class="fa fa-check"></i><b>14.5</b> Effectiveness 2</a></li>
<li class="chapter" data-level="14.6" data-path="remove-confounders-using-controls.html"><a href="remove-confounders-using-controls.html#exercise-4"><i class="fa fa-check"></i><b>14.6</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="remove-confounders-using-controls-reads.html"><a href="remove-confounders-using-controls-reads.html"><i class="fa fa-check"></i><b>15</b> Remove confounders using controls (Reads)</a></li>
<li class="divider"></li>
<li><a href="http://www.sanger.ac.uk/science/groups/hemberg-group" target="blank">Hemberg Lab, 2016</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Analysis of single-cell RNA-seq data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="normalization-for-library-size" class="section level1">
<h1><span class="header-section-number">12</span> Normalization for library size</h1>
<div id="introduction-4" class="section level2">
<h2><span class="header-section-number">12.1</span> Introduction</h2>
<p>In the previous chapter we identified important confounding factors and explanatory variables. scater allows one to account for these variables in subsequent statistical models or to condition them out using <code>normaliseExprs()</code>, if so desired. This can be done by providing a design matrix to <code>normaliseExprs()</code>. We are not covering this topic here, but you can try to do it yourself as an exercise.</p>
<p>Instead we will explore how simple size-factor normalisations correcting for library size can remove the effects of some of the confounders and explanatory variables.</p>
</div>
<div id="library-size-1" class="section level2">
<h2><span class="header-section-number">12.2</span> Library size</h2>
<p>Library sizes vary because of the various reasons: * scRNA-seq data is often sequenced on highly multiplexed platforms the total reads which are derived from each cell may differ substantially. * Protocols may differ in terms of their coverage of each transcript, their bias based on the average content of <strong>A/T</strong> nucleotides, or their ability to capture short transcripts.</p>
<p>Ideally, we would like to compensate for all of these differences and biases when comparing data from two different experiments.</p>
<p>Many methods to correct for library size have been developped for bulk RNA-seq and can be equally applied to scRNA-seq (eg. UQ, SF, CPM, RPKM, FPKM, TPM). Some quantification methods (eg. <a href="http://cole-trapnell-lab.github.io/cufflinks/">Cufflinks</a>, <a href="http://deweylab.github.io/RSEM/">RSEM</a>) incorporated library size when determining gene expression estimates thus do not require this normalization.</p>
<p>We will continue to work with the Blischak data that was used in the previous chapter and show how scater makes it easy to perform different types of size-factor normalizations.</p>
</div>
<div id="normalisations" class="section level2">
<h2><span class="header-section-number">12.3</span> Normalisations</h2>
<p>The simplest way to normalize this data is to convert it to counts per million (<strong>CPM</strong>) by dividing each column by its total then multiplying by 1,000,000. Note that spike-ins should be excluded from the calculation of total expression in order to correct for total cell RNA content, therefore we will only use endogenous genes. scater performs this normalisation by default, you can control it by changing <code>exprs_values</code> parameter to <code>&quot;exprs&quot;</code>.</p>
<p>Another method is called <strong>TMM</strong> is the weighted trimmed mean of M-values (to the reference) proposed by <a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2010-11-3-r25">Robinson and Oshlack (2010)</a>, where the weights are from the delta method on Binomial data.</p>
<p>Another very popular method <strong>RLE</strong> is the scaling factor method proposed by <a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2010-11-10-r106">Anders and Huber (2010)</a>. We call it “relative log expression”, as median library is calculated from the geometric mean of all columns and the median ratio of each sample to the median library is taken as the scale factor.</p>
<p>A similar to <strong>RLE</strong> the <strong>upperquartile</strong> is the upper-quartile normalization method of <a href="http://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-11-94">Bullard et al (2010)</a>, in which the scale factors are calculated from the 75% quantile of the counts for each library, after removing genes which are zero in all libraries. This idea is generalized here to allow scaling by any quantile of the distributions.</p>
<p>Let’s compare all these methods.</p>
<div id="raw" class="section level3">
<h3><span class="header-section-number">12.3.1</span> Raw</h3>
<div class="figure" style="text-align: center"><span id="fig:norm-pca-raw"></span>
<img src="scRNA-seq-course_files/figure-html/norm-pca-raw-1.png" alt="PCA plot of the blischak data" width="90%" />
<p class="caption">
Figure 12.1: PCA plot of the blischak data
</p>
</div>
</div>
<div id="cpm" class="section level3">
<h3><span class="header-section-number">12.3.2</span> CPM</h3>
<div class="figure" style="text-align: center"><span id="fig:norm-pca-cpm"></span>
<img src="scRNA-seq-course_files/figure-html/norm-pca-cpm-1.png" alt="PCA plot of the blischak data after CPM normalisation" width="90%" />
<p class="caption">
Figure 12.2: PCA plot of the blischak data after CPM normalisation
</p>
</div>
</div>
<div id="log2cpm" class="section level3">
<h3><span class="header-section-number">12.3.3</span> log2(CPM)</h3>
<div class="figure" style="text-align: center"><span id="fig:norm-pca-log2-cpm"></span>
<img src="scRNA-seq-course_files/figure-html/norm-pca-log2-cpm-1.png" alt="PCA plot of the blischak data after log2(CPM) normalisation" width="90%" />
<p class="caption">
Figure 12.3: PCA plot of the blischak data after log2(CPM) normalisation
</p>
</div>
</div>
<div id="tmm" class="section level3">
<h3><span class="header-section-number">12.3.4</span> TMM</h3>
<div class="figure" style="text-align: center"><span id="fig:norm-pca-tmm"></span>
<img src="scRNA-seq-course_files/figure-html/norm-pca-tmm-1.png" alt="PCA plot of the blischak data after TMM normalisation" width="90%" />
<p class="caption">
Figure 12.4: PCA plot of the blischak data after TMM normalisation
</p>
</div>
</div>
<div id="rle" class="section level3">
<h3><span class="header-section-number">12.3.5</span> RLE</h3>
<div class="figure" style="text-align: center"><span id="fig:norm-pca-rle"></span>
<img src="scRNA-seq-course_files/figure-html/norm-pca-rle-1.png" alt="PCA plot of the blischak data after RLE normalisation" width="90%" />
<p class="caption">
Figure 12.5: PCA plot of the blischak data after RLE normalisation
</p>
</div>
</div>
<div id="upperquantile" class="section level3">
<h3><span class="header-section-number">12.3.6</span> Upperquantile</h3>
<div class="figure" style="text-align: center"><span id="fig:norm-pca-uq"></span>
<img src="scRNA-seq-course_files/figure-html/norm-pca-uq-1.png" alt="PCA plot of the blischak data after UQ normalisation" width="90%" />
<p class="caption">
Figure 12.6: PCA plot of the blischak data after UQ normalisation
</p>
</div>
</div>
</div>
<div id="interpretation" class="section level2">
<h2><span class="header-section-number">12.4</span> Interpretation</h2>
<p>Clearly, only the CPM normalisation has reduced the effect of the number of detected genes and separated cells by individuals (though very weakly).</p>
</div>
<div id="other-methods" class="section level2">
<h2><span class="header-section-number">12.5</span> Other methods</h2>
<p>Some methods combine library size and fragment/gene length normalization such as:</p>
<ul>
<li><strong>RPKM</strong> - Reads Per Kilobase Million (for single-end sequencing)</li>
<li><strong>FPKM</strong> - Fragments Per Kilobase Million (same as <strong>RPKM</strong> but for paired-end sequencing, makes sure that paired ends mapped to the same fragment are not counted twice)</li>
<li><strong>TPM</strong> - Transcripts Per Kilobase Million (same as <strong>RPKM</strong>, but the order of normalizations is reversed - length first and sequencing depth second)</li>
</ul>
<p>These methods are not applicable to our dataset since the end of the transcript which contains the UMI was preferentially sequenced. Furthermore in general these should only be calculated using appropriate quantification software from aligned BAM files not from read counts since often only a portion of the entire gene/transcript is sequenced, not the entire length.</p>
<p>However, here we show how these normalisations can be calculated using scater. First, we need to find the effective transcript length in Kilobases. However, our dataset containes only gene IDs, therefore we will be using the gene lengths instead of transcripts. scater uses the <a href="https://bioconductor.org/packages/release/bioc/html/biomaRt.html">biomaRt</a> package, which allows one to annotate genes by other attributes:</p>
<p>Some of the genes were not annotated, therefore we filter them out:</p>
<p>Now we compute the effective gene length in Kilobases by using the <code>end_position</code> and <code>start_position</code> fields:</p>
<p>Now we are ready to perform the normalisations:</p>
Plot the results as a PCA plot:
<div class="figure" style="text-align: center"><span id="fig:norm-pca-fpkm"></span>
<img src="scRNA-seq-course_files/figure-html/norm-pca-fpkm-1.png" alt="PCA plot of the blischak data after FPKM normalisation" width="90%" />
<p class="caption">
Figure 12.7: PCA plot of the blischak data after FPKM normalisation
</p>
</div>
<p>TPM normalisation produce a zero-matrix, we are not sure why, it maybe a bug in scater.</p>
</div>
<div id="visualize-genes" class="section level2">
<h2><span class="header-section-number">12.6</span> Visualize genes</h2>
<p>Now after the normalisation we are ready to visualise the gene expression:</p>
<div id="raw-1" class="section level3">
<h3><span class="header-section-number">12.6.1</span> Raw</h3>
<div class="figure" style="text-align: center"><span id="fig:norm-genes-raw"></span>
<img src="scRNA-seq-course_files/figure-html/norm-genes-raw-1.png" alt="Expression of the first 6 genes of the blischak data" width="90%" />
<p class="caption">
Figure 12.8: Expression of the first 6 genes of the blischak data
</p>
</div>
</div>
<div id="cpm-1" class="section level3">
<h3><span class="header-section-number">12.6.2</span> CPM</h3>
<div class="figure" style="text-align: center"><span id="fig:norm-genes-cpm"></span>
<img src="scRNA-seq-course_files/figure-html/norm-genes-cpm-1.png" alt="Expression of the first 6 genes of the blischak data after the CPM normalisation" width="90%" />
<p class="caption">
Figure 12.9: Expression of the first 6 genes of the blischak data after the CPM normalisation
</p>
</div>
</div>
<div id="log2cpm-1" class="section level3">
<h3><span class="header-section-number">12.6.3</span> log2(CPM)</h3>
<div class="figure" style="text-align: center"><span id="fig:norm-genes-log2-cpm"></span>
<img src="scRNA-seq-course_files/figure-html/norm-genes-log2-cpm-1.png" alt="Expression of the first 6 genes of the blischak data after the log2(CPM) normalisation" width="90%" />
<p class="caption">
Figure 12.10: Expression of the first 6 genes of the blischak data after the log2(CPM) normalisation
</p>
</div>
</div>
<div id="upperquantile-1" class="section level3">
<h3><span class="header-section-number">12.6.4</span> Upperquantile</h3>
<div class="figure" style="text-align: center"><span id="fig:norm-genes-UQ"></span>
<img src="scRNA-seq-course_files/figure-html/norm-genes-UQ-1.png" alt="Expression of the first 6 genes of the blischak data after the UQ normalisation" width="90%" />
<p class="caption">
Figure 12.11: Expression of the first 6 genes of the blischak data after the UQ normalisation
</p>
</div>
</div>
<div id="fpkm" class="section level3">
<h3><span class="header-section-number">12.6.5</span> FPKM</h3>
<div class="figure" style="text-align: center"><span id="fig:norm-genes-fpkm"></span>
<img src="scRNA-seq-course_files/figure-html/norm-genes-fpkm-1.png" alt="Expression of the first 6 genes of the blischak data after the FPKM normalisation" width="90%" />
<p class="caption">
Figure 12.12: Expression of the first 6 genes of the blischak data after the FPKM normalisation
</p>
</div>
</div>
<div id="tpm" class="section level3">
<h3><span class="header-section-number">12.6.6</span> TPM</h3>
<div class="figure" style="text-align: center"><span id="fig:norm-genes-tpm"></span>
<img src="scRNA-seq-course_files/figure-html/norm-genes-tpm-1.png" alt="Expression of the first 6 genes of the blischak data after the TPM normalisation" width="90%" />
<p class="caption">
Figure 12.13: Expression of the first 6 genes of the blischak data after the TPM normalisation
</p>
</div>
</div>
</div>
<div id="exercise-3" class="section level2">
<h2><span class="header-section-number">12.7</span> Exercise</h2>
<p>Perform the same analysis with read counts of the Blischak data. Use <code>blischak/reads.rds</code> file to load the reads SCESet object. Once you have finished please compare your results to ours (next chapter).</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="confounding-factors-reads.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="normalization-for-library-size-reads.html" class="navigation navigation-next " aria-label="Next page""><i class="fa fa-angle-right"></i></a>

<script src="libs/gitbook-2.6.7/js/app.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/wikiselev/bookdown-demo/edit/master/11-exprs-norm.Rmd",
"text": "Edit"
},
"download": null,
"toc": {
"collapse": "subsection"
},
"search": true
});
});
</script>


</body>

</html>
