<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Analysis of single cell RNA-seq data</title>
  <meta content="text/html; charset=UTF-8" http-equiv="Content-Type">
  <meta name="description" content="Analysis of single cell RNA-seq data">
  <meta name="generator" content="bookdown 0.3 and GitBook 2.6.7">

  <meta property="og:title" content="Analysis of single cell RNA-seq data" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Analysis of single cell RNA-seq data" />
  
  
  

<meta name="author" content="Vladimir Kiselev, Tallulah Andrews, Davis McCarthy and Martin Hemberg">


<meta name="date" content="2017-03-10">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="clustering-introduction.html">
<link rel="next" href="identification-of-important-genes.html">

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<!-- for Facebook -->  
<meta property="og:url" content="http://hemberg-lab.github.io/scRNA.seq.course/" />
<meta property="og:description" content="In this course we will be surveying the existing problems as well as the available computational and statistical frameworks available for the analysis of scRNA-seq. The course is taught through the University of Cambridge Bioinformatics training unit, but the material found on these pages is meant to be used for anyone interested in learning about computational analysis of scRNA-seq data." />
<meta property="og:image" content="http://hemberg-lab.github.io/scRNA.seq.course/figures/RNA-Seq_workflow-5.pdf.jpg" />

<!-- for Twitter -->          
<meta name="twitter:card" content="summary_large_image" />
<meta name="twitter:title" content="Analysis of single-cell RNA-seq data" />
<meta name="twitter:description" content="In this course we will be surveying the existing problems as well as the available computational and statistical frameworks available for the analysis of scRNA-seq. The course is taught through the University of Cambridge Bioinformatics training unit, but the material found on these pages is meant to be used for anyone interested in learning about computational analysis of scRNA-seq data." />
<meta name="twitter:image" content="http://hemberg-lab.github.io/scRNA.seq.course/figures/RNA-Seq_workflow-5.pdf.jpg" />

<!-- Google Analytics -->
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-71525309-1', 'auto');
  ga('send', 'pageview');

</script>


<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>


  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="index.html">Table of Contents</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About the course</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#video"><i class="fa fa-check"></i><b>1.1</b> Video</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#registration"><i class="fa fa-check"></i><b>1.2</b> Registration</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#github"><i class="fa fa-check"></i><b>1.3</b> GitHub</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>1.4</b> License</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#prerequisites"><i class="fa fa-check"></i><b>1.5</b> Prerequisites</a></li>
<li class="chapter" data-level="1.6" data-path="index.html"><a href="index.html#contact"><i class="fa fa-check"></i><b>1.6</b> Contact</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="technical-requirements.html"><a href="technical-requirements.html"><i class="fa fa-check"></i><b>2</b> Technical requirements</a><ul>
<li class="chapter" data-level="2.1" data-path="technical-requirements.html"><a href="technical-requirements.html#general"><i class="fa fa-check"></i><b>2.1</b> General</a></li>
<li class="chapter" data-level="2.2" data-path="technical-requirements.html"><a href="technical-requirements.html#plotting"><i class="fa fa-check"></i><b>2.2</b> Plotting</a></li>
<li class="chapter" data-level="2.3" data-path="technical-requirements.html"><a href="technical-requirements.html#qc-and-normalisation"><i class="fa fa-check"></i><b>2.3</b> QC and normalisation</a></li>
<li class="chapter" data-level="2.4" data-path="technical-requirements.html"><a href="technical-requirements.html#clustering"><i class="fa fa-check"></i><b>2.4</b> Clustering</a></li>
<li class="chapter" data-level="2.5" data-path="technical-requirements.html"><a href="technical-requirements.html#dropouts"><i class="fa fa-check"></i><b>2.5</b> Dropouts</a></li>
<li class="chapter" data-level="2.6" data-path="technical-requirements.html"><a href="technical-requirements.html#pseudotime"><i class="fa fa-check"></i><b>2.6</b> Pseudotime</a></li>
<li class="chapter" data-level="2.7" data-path="technical-requirements.html"><a href="technical-requirements.html#differential-expression"><i class="fa fa-check"></i><b>2.7</b> Differential Expression</a></li>
<li class="chapter" data-level="2.8" data-path="technical-requirements.html"><a href="technical-requirements.html#extra-tools"><i class="fa fa-check"></i><b>2.8</b> Extra tools</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html"><i class="fa fa-check"></i><b>3</b> Introduction to single-cell RNA-seq</a><ul>
<li class="chapter" data-level="3.1" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#bulk-rna-seq"><i class="fa fa-check"></i><b>3.1</b> Bulk RNA-seq</a></li>
<li class="chapter" data-level="3.2" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#scrna-seq"><i class="fa fa-check"></i><b>3.2</b> scRNA-seq</a></li>
<li class="chapter" data-level="3.3" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#workflow"><i class="fa fa-check"></i><b>3.3</b> Workflow</a></li>
<li class="chapter" data-level="3.4" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#computational-analysis"><i class="fa fa-check"></i><b>3.4</b> Computational Analysis</a></li>
<li class="chapter" data-level="3.5" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#challenges"><i class="fa fa-check"></i><b>3.5</b> Challenges</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="overview-of-experimental-methods-for-generating-scrna-seq-data.html"><a href="overview-of-experimental-methods-for-generating-scrna-seq-data.html"><i class="fa fa-check"></i><b>4</b> Overview of experimental methods for generating scRNA-seq data</a></li>
<li class="chapter" data-level="5" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html"><i class="fa fa-check"></i><b>5</b> Construction of expression matrix</a><ul>
<li class="chapter" data-level="5.1" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#reads-qc"><i class="fa fa-check"></i><b>5.1</b> Reads QC</a></li>
<li class="chapter" data-level="5.2" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#reads-alignment"><i class="fa fa-check"></i><b>5.2</b> Reads alignment</a></li>
<li class="chapter" data-level="5.3" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#alignment-example"><i class="fa fa-check"></i><b>5.3</b> Alignment example</a></li>
<li class="chapter" data-level="5.4" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#mapping-qc"><i class="fa fa-check"></i><b>5.4</b> Mapping QC</a></li>
<li class="chapter" data-level="5.5" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#reads-quantification"><i class="fa fa-check"></i><b>5.5</b> Reads quantification</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="unique-molecular-identifiers-umis.html"><a href="unique-molecular-identifiers-umis.html"><i class="fa fa-check"></i><b>6</b> Unique Molecular Identifiers (UMIs)</a><ul>
<li class="chapter" data-level="6.1" data-path="unique-molecular-identifiers-umis.html"><a href="unique-molecular-identifiers-umis.html#introduction"><i class="fa fa-check"></i><b>6.1</b> Introduction</a></li>
<li class="chapter" data-level="6.2" data-path="unique-molecular-identifiers-umis.html"><a href="unique-molecular-identifiers-umis.html#mapping-barcodes"><i class="fa fa-check"></i><b>6.2</b> Mapping Barcodes</a></li>
<li class="chapter" data-level="6.3" data-path="unique-molecular-identifiers-umis.html"><a href="unique-molecular-identifiers-umis.html#counting-barcodes"><i class="fa fa-check"></i><b>6.3</b> Counting Barcodes</a></li>
<li class="chapter" data-level="6.4" data-path="unique-molecular-identifiers-umis.html"><a href="unique-molecular-identifiers-umis.html#correcting-for-errors"><i class="fa fa-check"></i><b>6.4</b> Correcting for Errors</a></li>
<li class="chapter" data-level="6.5" data-path="unique-molecular-identifiers-umis.html"><a href="unique-molecular-identifiers-umis.html#downstream-analysis"><i class="fa fa-check"></i><b>6.5</b> Downstream Analysis</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="scater-package.html"><a href="scater-package.html"><i class="fa fa-check"></i><b>7</b> scater package</a><ul>
<li class="chapter" data-level="7.1" data-path="scater-package.html"><a href="scater-package.html#introduction-1"><i class="fa fa-check"></i><b>7.1</b> Introduction</a></li>
<li class="chapter" data-level="7.2" data-path="scater-package.html"><a href="scater-package.html#scater-workflow"><i class="fa fa-check"></i><b>7.2</b> scater workflow</a></li>
<li class="chapter" data-level="7.3" data-path="scater-package.html"><a href="scater-package.html#terminology"><i class="fa fa-check"></i><b>7.3</b> Terminology</a></li>
<li class="chapter" data-level="7.4" data-path="scater-package.html"><a href="scater-package.html#sceset-class"><i class="fa fa-check"></i><b>7.4</b> SCESet class</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html"><i class="fa fa-check"></i><b>8</b> Expression QC (UMI)</a><ul>
<li class="chapter" data-level="8.1" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#introduction-2"><i class="fa fa-check"></i><b>8.1</b> Introduction</a></li>
<li class="chapter" data-level="8.2" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#blischak-dataset"><i class="fa fa-check"></i><b>8.2</b> Blischak dataset</a></li>
<li class="chapter" data-level="8.3" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#cell-qc"><i class="fa fa-check"></i><b>8.3</b> Cell QC</a><ul>
<li class="chapter" data-level="8.3.1" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#library-size"><i class="fa fa-check"></i><b>8.3.1</b> Library size</a></li>
<li class="chapter" data-level="8.3.2" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#detected-genes-1"><i class="fa fa-check"></i><b>8.3.2</b> Detected genes (1)</a></li>
<li class="chapter" data-level="8.3.3" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#erccs-and-mts"><i class="fa fa-check"></i><b>8.3.3</b> ERCCs and MTs</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#cell-filtering"><i class="fa fa-check"></i><b>8.4</b> Cell filtering</a><ul>
<li class="chapter" data-level="8.4.1" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#manual"><i class="fa fa-check"></i><b>8.4.1</b> Manual</a></li>
<li class="chapter" data-level="8.4.2" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#default-thresholds"><i class="fa fa-check"></i><b>8.4.2</b> Default Thresholds</a></li>
<li class="chapter" data-level="8.4.3" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#automatic"><i class="fa fa-check"></i><b>8.4.3</b> Automatic</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#compare-filterings"><i class="fa fa-check"></i><b>8.5</b> Compare filterings</a></li>
<li class="chapter" data-level="8.6" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#gene-analysis"><i class="fa fa-check"></i><b>8.6</b> Gene analysis</a><ul>
<li class="chapter" data-level="8.6.1" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#gene-expression"><i class="fa fa-check"></i><b>8.6.1</b> Gene expression</a></li>
<li class="chapter" data-level="8.6.2" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#gene-filtering"><i class="fa fa-check"></i><b>8.6.2</b> Gene filtering</a></li>
</ul></li>
<li class="chapter" data-level="8.7" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#save-the-data"><i class="fa fa-check"></i><b>8.7</b> Save the data</a></li>
<li class="chapter" data-level="8.8" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#big-exercise"><i class="fa fa-check"></i><b>8.8</b> Big Exercise</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="expression-qc-reads.html"><a href="expression-qc-reads.html"><i class="fa fa-check"></i><b>9</b> Expression QC (Reads)</a></li>
<li class="chapter" data-level="10" data-path="data-visualization.html"><a href="data-visualization.html"><i class="fa fa-check"></i><b>10</b> Data visualization</a><ul>
<li class="chapter" data-level="10.1" data-path="data-visualization.html"><a href="data-visualization.html#introduction-3"><i class="fa fa-check"></i><b>10.1</b> Introduction</a></li>
<li class="chapter" data-level="10.2" data-path="data-visualization.html"><a href="data-visualization.html#visual-pca"><i class="fa fa-check"></i><b>10.2</b> PCA plot</a><ul>
<li class="chapter" data-level="10.2.1" data-path="data-visualization.html"><a href="data-visualization.html#before-qc"><i class="fa fa-check"></i><b>10.2.1</b> Before QC</a></li>
<li class="chapter" data-level="10.2.2" data-path="data-visualization.html"><a href="data-visualization.html#after-qc"><i class="fa fa-check"></i><b>10.2.2</b> After QC</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="data-visualization.html"><a href="data-visualization.html#visual-tsne"><i class="fa fa-check"></i><b>10.3</b> tSNE map</a><ul>
<li class="chapter" data-level="10.3.1" data-path="data-visualization.html"><a href="data-visualization.html#before-qc-1"><i class="fa fa-check"></i><b>10.3.1</b> Before QC</a></li>
<li class="chapter" data-level="10.3.2" data-path="data-visualization.html"><a href="data-visualization.html#after-qc-1"><i class="fa fa-check"></i><b>10.3.2</b> After QC</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="data-visualization.html"><a href="data-visualization.html#big-exercise-1"><i class="fa fa-check"></i><b>10.4</b> Big Exercise</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="data-visualization-reads.html"><a href="data-visualization-reads.html"><i class="fa fa-check"></i><b>11</b> Data visualization (Reads)</a></li>
<li class="chapter" data-level="12" data-path="confounding-factors.html"><a href="confounding-factors.html"><i class="fa fa-check"></i><b>12</b> Confounding factors</a><ul>
<li class="chapter" data-level="12.1" data-path="confounding-factors.html"><a href="confounding-factors.html#introduction-4"><i class="fa fa-check"></i><b>12.1</b> Introduction</a></li>
<li class="chapter" data-level="12.2" data-path="confounding-factors.html"><a href="confounding-factors.html#correlations-with-pcs"><i class="fa fa-check"></i><b>12.2</b> Correlations with PCs</a><ul>
<li class="chapter" data-level="12.2.1" data-path="confounding-factors.html"><a href="confounding-factors.html#detected-genes"><i class="fa fa-check"></i><b>12.2.1</b> Detected genes</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="confounding-factors.html"><a href="confounding-factors.html#explanatory-variables"><i class="fa fa-check"></i><b>12.3</b> Explanatory variables</a></li>
<li class="chapter" data-level="12.4" data-path="confounding-factors.html"><a href="confounding-factors.html#other-confounders"><i class="fa fa-check"></i><b>12.4</b> Other confounders</a></li>
<li class="chapter" data-level="12.5" data-path="confounding-factors.html"><a href="confounding-factors.html#exercise"><i class="fa fa-check"></i><b>12.5</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="confounding-factors-reads.html"><a href="confounding-factors-reads.html"><i class="fa fa-check"></i><b>13</b> Confounding factors (Reads)</a></li>
<li class="chapter" data-level="14" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html"><i class="fa fa-check"></i><b>14</b> Normalization for library size</a><ul>
<li class="chapter" data-level="14.1" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#introduction-5"><i class="fa fa-check"></i><b>14.1</b> Introduction</a></li>
<li class="chapter" data-level="14.2" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#library-size-1"><i class="fa fa-check"></i><b>14.2</b> Library size</a></li>
<li class="chapter" data-level="14.3" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#normalisations"><i class="fa fa-check"></i><b>14.3</b> Normalisations</a><ul>
<li class="chapter" data-level="14.3.1" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#raw"><i class="fa fa-check"></i><b>14.3.1</b> Raw</a></li>
<li class="chapter" data-level="14.3.2" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#cpm"><i class="fa fa-check"></i><b>14.3.2</b> CPM</a></li>
<li class="chapter" data-level="14.3.3" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#tmm"><i class="fa fa-check"></i><b>14.3.3</b> TMM</a></li>
<li class="chapter" data-level="14.3.4" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#scran"><i class="fa fa-check"></i><b>14.3.4</b> scran</a></li>
<li class="chapter" data-level="14.3.5" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#size-factor-rle"><i class="fa fa-check"></i><b>14.3.5</b> Size-factor (RLE)</a></li>
<li class="chapter" data-level="14.3.6" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#upperquantile"><i class="fa fa-check"></i><b>14.3.6</b> Upperquantile</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#downsampling"><i class="fa fa-check"></i><b>14.4</b> Downsampling</a></li>
<li class="chapter" data-level="14.5" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#normalizing-for-genetranscript-length"><i class="fa fa-check"></i><b>14.5</b> Normalizing for gene/transcript length</a></li>
<li class="chapter" data-level="14.6" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#exercise-1"><i class="fa fa-check"></i><b>14.6</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="normalization-for-library-size-reads.html"><a href="normalization-for-library-size-reads.html"><i class="fa fa-check"></i><b>15</b> Normalization for library size (Reads)</a></li>
<li class="chapter" data-level="16" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html"><i class="fa fa-check"></i><b>16</b> Dealing with confounders</a><ul>
<li class="chapter" data-level="16.1" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html#introduction-6"><i class="fa fa-check"></i><b>16.1</b> Introduction</a></li>
<li class="chapter" data-level="16.2" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html#remove-unwanted-variation"><i class="fa fa-check"></i><b>16.2</b> Remove Unwanted Variation</a><ul>
<li class="chapter" data-level="16.2.1" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html#ruvg"><i class="fa fa-check"></i><b>16.2.1</b> RUVg</a></li>
<li class="chapter" data-level="16.2.2" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html#ruvs"><i class="fa fa-check"></i><b>16.2.2</b> RUVs</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html#effectiveness-1"><i class="fa fa-check"></i><b>16.3</b> Effectiveness 1</a></li>
<li class="chapter" data-level="16.4" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html#effectiveness-2"><i class="fa fa-check"></i><b>16.4</b> Effectiveness 2</a></li>
<li class="chapter" data-level="16.5" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html#effectiveness-3"><i class="fa fa-check"></i><b>16.5</b> Effectiveness 3</a><ul>
<li class="chapter" data-level="16.5.1" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html#de-raw-counts"><i class="fa fa-check"></i><b>16.5.1</b> DE (raw counts)</a></li>
<li class="chapter" data-level="16.5.2" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html#de-ruvg-k-2"><i class="fa fa-check"></i><b>16.5.2</b> DE (RUVg, k = 2)</a></li>
<li class="chapter" data-level="16.5.3" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html#de-ruvs-k-2"><i class="fa fa-check"></i><b>16.5.3</b> DE (RUVs, k = 2)</a></li>
</ul></li>
<li class="chapter" data-level="16.6" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html#exercise-2"><i class="fa fa-check"></i><b>16.6</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="dealing-with-confounders-reads.html"><a href="dealing-with-confounders-reads.html"><i class="fa fa-check"></i><b>17</b> Dealing with confounders (Reads)</a><ul>
<li class="chapter" data-level="17.1" data-path="dealing-with-confounders-reads.html"><a href="dealing-with-confounders-reads.html#remove-unwanted-variation-1"><i class="fa fa-check"></i><b>17.1</b> Remove Unwanted Variation</a><ul>
<li class="chapter" data-level="17.1.1" data-path="dealing-with-confounders-reads.html"><a href="dealing-with-confounders-reads.html#ruvg-1"><i class="fa fa-check"></i><b>17.1.1</b> RUVg</a></li>
<li class="chapter" data-level="17.1.2" data-path="dealing-with-confounders-reads.html"><a href="dealing-with-confounders-reads.html#ruvs-1"><i class="fa fa-check"></i><b>17.1.2</b> RUVs</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="dealing-with-confounders-reads.html"><a href="dealing-with-confounders-reads.html#effectiveness-1-1"><i class="fa fa-check"></i><b>17.2</b> Effectiveness 1</a></li>
<li class="chapter" data-level="17.3" data-path="dealing-with-confounders-reads.html"><a href="dealing-with-confounders-reads.html#effectiveness-2-1"><i class="fa fa-check"></i><b>17.3</b> Effectiveness 2</a></li>
<li class="chapter" data-level="17.4" data-path="dealing-with-confounders-reads.html"><a href="dealing-with-confounders-reads.html#effectiveness-3-1"><i class="fa fa-check"></i><b>17.4</b> Effectiveness 3</a><ul>
<li class="chapter" data-level="17.4.1" data-path="dealing-with-confounders-reads.html"><a href="dealing-with-confounders-reads.html#de-raw-counts-1"><i class="fa fa-check"></i><b>17.4.1</b> DE (raw counts)</a></li>
<li class="chapter" data-level="17.4.2" data-path="dealing-with-confounders-reads.html"><a href="dealing-with-confounders-reads.html#de-ruvg-k-2-1"><i class="fa fa-check"></i><b>17.4.2</b> DE (RUVg, k = 2)</a></li>
<li class="chapter" data-level="17.4.3" data-path="dealing-with-confounders-reads.html"><a href="dealing-with-confounders-reads.html#de-ruvs-k-2-1"><i class="fa fa-check"></i><b>17.4.3</b> DE (RUVs, k = 2)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="clustering-introduction.html"><a href="clustering-introduction.html"><i class="fa fa-check"></i><b>18</b> Clustering Introduction</a><ul>
<li class="chapter" data-level="18.1" data-path="clustering-introduction.html"><a href="clustering-introduction.html#introduction-7"><i class="fa fa-check"></i><b>18.1</b> Introduction</a></li>
<li class="chapter" data-level="18.2" data-path="clustering-introduction.html"><a href="clustering-introduction.html#dimensionality-reductions"><i class="fa fa-check"></i><b>18.2</b> Dimensionality reductions</a><ul>
<li class="chapter" data-level="18.2.1" data-path="clustering-introduction.html"><a href="clustering-introduction.html#pca"><i class="fa fa-check"></i><b>18.2.1</b> PCA</a></li>
<li class="chapter" data-level="18.2.2" data-path="clustering-introduction.html"><a href="clustering-introduction.html#spectral"><i class="fa fa-check"></i><b>18.2.2</b> Spectral</a></li>
<li class="chapter" data-level="18.2.3" data-path="clustering-introduction.html"><a href="clustering-introduction.html#tsne"><i class="fa fa-check"></i><b>18.2.3</b> tSNE</a></li>
</ul></li>
<li class="chapter" data-level="18.3" data-path="clustering-introduction.html"><a href="clustering-introduction.html#clustering-methods"><i class="fa fa-check"></i><b>18.3</b> Clustering methods</a><ul>
<li class="chapter" data-level="18.3.1" data-path="clustering-introduction.html"><a href="clustering-introduction.html#hierarchical-clustering"><i class="fa fa-check"></i><b>18.3.1</b> Hierarchical clustering</a></li>
<li class="chapter" data-level="18.3.2" data-path="clustering-introduction.html"><a href="clustering-introduction.html#k-means"><i class="fa fa-check"></i><b>18.3.2</b> k-means</a></li>
<li class="chapter" data-level="18.3.3" data-path="clustering-introduction.html"><a href="clustering-introduction.html#graph-based-methods"><i class="fa fa-check"></i><b>18.3.3</b> Graph-based methods</a></li>
</ul></li>
<li class="chapter" data-level="18.4" data-path="clustering-introduction.html"><a href="clustering-introduction.html#challenges-in-clustering"><i class="fa fa-check"></i><b>18.4</b> Challenges in clustering</a></li>
<li class="chapter" data-level="18.5" data-path="clustering-introduction.html"><a href="clustering-introduction.html#tools-for-scrna-seq-data"><i class="fa fa-check"></i><b>18.5</b> Tools for scRNA-seq data</a><ul>
<li class="chapter" data-level="18.5.1" data-path="clustering-introduction.html"><a href="clustering-introduction.html#sincera"><i class="fa fa-check"></i><b>18.5.1</b> <a href="https://research.cchmc.org/pbge/sincera.html">SINCERA</a></a></li>
<li class="chapter" data-level="18.5.2" data-path="clustering-introduction.html"><a href="clustering-introduction.html#pcareduce"><i class="fa fa-check"></i><b>18.5.2</b> <a href="https://github.com/JustinaZ/pcaReduce">pcaReduce</a></a></li>
<li class="chapter" data-level="18.5.3" data-path="clustering-introduction.html"><a href="clustering-introduction.html#sc3"><i class="fa fa-check"></i><b>18.5.3</b> <a href="http://bioconductor.org/packages/SC3/">SC3</a></a></li>
<li class="chapter" data-level="18.5.4" data-path="clustering-introduction.html"><a href="clustering-introduction.html#tsne-k-means"><i class="fa fa-check"></i><b>18.5.4</b> tSNE + k-means</a></li>
<li class="chapter" data-level="18.5.5" data-path="clustering-introduction.html"><a href="clustering-introduction.html#seurat"><i class="fa fa-check"></i><b>18.5.5</b> <a href="https://github.com/Puriney/seurat">SEURAT</a></a></li>
<li class="chapter" data-level="18.5.6" data-path="clustering-introduction.html"><a href="clustering-introduction.html#snn-cliq"><i class="fa fa-check"></i><b>18.5.6</b> <a href="http://bioinfo.uncc.edu/SNNCliq/">SNN-Cliq</a></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="19" data-path="clust-methods.html"><a href="clust-methods.html"><i class="fa fa-check"></i><b>19</b> Clustering example</a><ul>
<li class="chapter" data-level="19.1" data-path="clust-methods.html"><a href="clust-methods.html#patient-dataset"><i class="fa fa-check"></i><b>19.1</b> Patient dataset</a></li>
<li class="chapter" data-level="19.2" data-path="clust-methods.html"><a href="clust-methods.html#sc3-1"><i class="fa fa-check"></i><b>19.2</b> SC3</a></li>
<li class="chapter" data-level="19.3" data-path="clust-methods.html"><a href="clust-methods.html#pcareduce-1"><i class="fa fa-check"></i><b>19.3</b> pcaReduce</a></li>
<li class="chapter" data-level="19.4" data-path="clust-methods.html"><a href="clust-methods.html#tsne-kmeans"><i class="fa fa-check"></i><b>19.4</b> tSNE + kmeans</a></li>
<li class="chapter" data-level="19.5" data-path="clust-methods.html"><a href="clust-methods.html#snn-cliq-1"><i class="fa fa-check"></i><b>19.5</b> SNN-Cliq</a></li>
<li class="chapter" data-level="19.6" data-path="clust-methods.html"><a href="clust-methods.html#sincera-1"><i class="fa fa-check"></i><b>19.6</b> SINCERA</a></li>
<li class="chapter" data-level="19.7" data-path="clust-methods.html"><a href="clust-methods.html#seurat-1"><i class="fa fa-check"></i><b>19.7</b> SEURAT</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="identification-of-important-genes.html"><a href="identification-of-important-genes.html"><i class="fa fa-check"></i><b>20</b> Identification of important genes</a><ul>
<li class="chapter" data-level="20.1" data-path="identification-of-important-genes.html"><a href="identification-of-important-genes.html#fitting-the-models"><i class="fa fa-check"></i><b>20.1</b> Fitting the models</a></li>
<li class="chapter" data-level="20.2" data-path="identification-of-important-genes.html"><a href="identification-of-important-genes.html#the-michaelis-menten-equation"><i class="fa fa-check"></i><b>20.2</b> The Michaelis-Menten Equation</a></li>
<li class="chapter" data-level="20.3" data-path="identification-of-important-genes.html"><a href="identification-of-important-genes.html#right-outliers"><i class="fa fa-check"></i><b>20.3</b> Right outliers</a></li>
<li class="chapter" data-level="20.4" data-path="identification-of-important-genes.html"><a href="identification-of-important-genes.html#validation-of-de-results"><i class="fa fa-check"></i><b>20.4</b> Validation of DE results</a></li>
<li class="chapter" data-level="20.5" data-path="identification-of-important-genes.html"><a href="identification-of-important-genes.html#comparing-m3drop-to-other-methods"><i class="fa fa-check"></i><b>20.5</b> Comparing M3Drop to other methods</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="pseudotime-analysis.html"><a href="pseudotime-analysis.html"><i class="fa fa-check"></i><b>21</b> Pseudotime analysis</a><ul>
<li class="chapter" data-level="21.1" data-path="pseudotime-analysis.html"><a href="pseudotime-analysis.html#tscan"><i class="fa fa-check"></i><b>21.1</b> TSCAN</a></li>
<li class="chapter" data-level="21.2" data-path="pseudotime-analysis.html"><a href="pseudotime-analysis.html#monocle"><i class="fa fa-check"></i><b>21.2</b> monocle</a></li>
<li class="chapter" data-level="21.3" data-path="pseudotime-analysis.html"><a href="pseudotime-analysis.html#diffusion-maps"><i class="fa fa-check"></i><b>21.3</b> Diffusion maps</a></li>
<li class="chapter" data-level="21.4" data-path="pseudotime-analysis.html"><a href="pseudotime-analysis.html#comparison-of-the-methods"><i class="fa fa-check"></i><b>21.4</b> Comparison of the methods</a></li>
<li class="chapter" data-level="21.5" data-path="pseudotime-analysis.html"><a href="pseudotime-analysis.html#expression-of-genes-through-time"><i class="fa fa-check"></i><b>21.5</b> Expression of genes through time</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="differential-expression-de-analysis.html"><a href="differential-expression-de-analysis.html"><i class="fa fa-check"></i><b>22</b> Differential Expression (DE) analysis</a><ul>
<li class="chapter" data-level="22.1" data-path="differential-expression-de-analysis.html"><a href="differential-expression-de-analysis.html#bulk-rna-seq-1"><i class="fa fa-check"></i><b>22.1</b> Bulk RNA-seq</a></li>
<li class="chapter" data-level="22.2" data-path="differential-expression-de-analysis.html"><a href="differential-expression-de-analysis.html#single-cell-rna-seq"><i class="fa fa-check"></i><b>22.2</b> Single cell RNA-seq</a></li>
<li class="chapter" data-level="22.3" data-path="differential-expression-de-analysis.html"><a href="differential-expression-de-analysis.html#differences-in-distribution"><i class="fa fa-check"></i><b>22.3</b> Differences in Distribution</a></li>
<li class="chapter" data-level="22.4" data-path="differential-expression-de-analysis.html"><a href="differential-expression-de-analysis.html#models-of-single-cell-rnaseq-data"><i class="fa fa-check"></i><b>22.4</b> Models of single-cell RNASeq data</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html"><i class="fa fa-check"></i><b>23</b> DE in a real dataset</a><ul>
<li class="chapter" data-level="23.1" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#introduction-8"><i class="fa fa-check"></i><b>23.1</b> Introduction</a></li>
<li class="chapter" data-level="23.2" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#kolmogorov-smirnov-test"><i class="fa fa-check"></i><b>23.2</b> Kolmogorov-Smirnov test</a><ul>
<li class="chapter" data-level="23.2.1" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#evaluating-accuracy"><i class="fa fa-check"></i><b>23.2.1</b> Evaluating Accuracy</a></li>
</ul></li>
<li class="chapter" data-level="23.3" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#wilcoxmann-whitney-u-test"><i class="fa fa-check"></i><b>23.3</b> Wilcox/Mann-Whitney-U Test</a></li>
<li class="chapter" data-level="23.4" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#edger"><i class="fa fa-check"></i><b>23.4</b> edgeR</a></li>
<li class="chapter" data-level="23.5" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#monocle-1"><i class="fa fa-check"></i><b>23.5</b> Monocle</a></li>
<li class="chapter" data-level="23.6" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#mast"><i class="fa fa-check"></i><b>23.6</b> MAST</a></li>
<li class="chapter" data-level="23.7" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#slow-methods-1h-to-run"><i class="fa fa-check"></i><b>23.7</b> Slow Methods (&gt;1h to run)</a></li>
<li class="chapter" data-level="23.8" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#bpsc"><i class="fa fa-check"></i><b>23.8</b> BPSC</a></li>
<li class="chapter" data-level="23.9" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#scde"><i class="fa fa-check"></i><b>23.9</b> SCDE</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="ideal-scrnaseq-pipeline-as-of-oct-2016.html"><a href="ideal-scrnaseq-pipeline-as-of-oct-2016.html"><i class="fa fa-check"></i><b>24</b> “Ideal” scRNAseq pipeline (as of Oct 2016)</a><ul>
<li class="chapter" data-level="24.1" data-path="ideal-scrnaseq-pipeline-as-of-oct-2016.html"><a href="ideal-scrnaseq-pipeline-as-of-oct-2016.html#experimental-design"><i class="fa fa-check"></i><b>24.1</b> Experimental Design</a></li>
<li class="chapter" data-level="24.2" data-path="ideal-scrnaseq-pipeline-as-of-oct-2016.html"><a href="ideal-scrnaseq-pipeline-as-of-oct-2016.html#processing-reads"><i class="fa fa-check"></i><b>24.2</b> Processing Reads</a></li>
<li class="chapter" data-level="24.3" data-path="ideal-scrnaseq-pipeline-as-of-oct-2016.html"><a href="ideal-scrnaseq-pipeline-as-of-oct-2016.html#preparing-expression-matrix"><i class="fa fa-check"></i><b>24.3</b> Preparing Expression Matrix</a></li>
<li class="chapter" data-level="24.4" data-path="ideal-scrnaseq-pipeline-as-of-oct-2016.html"><a href="ideal-scrnaseq-pipeline-as-of-oct-2016.html#biological-interpretation"><i class="fa fa-check"></i><b>24.4</b> Biological Interpretation</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="advanced-exercises.html"><a href="advanced-exercises.html"><i class="fa fa-check"></i><b>25</b> Advanced exercises</a></li>
<li class="chapter" data-level="26" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>26</b> Resources</a><ul>
<li class="chapter" data-level="26.1" data-path="resources.html"><a href="resources.html#scrna-seq-protocols"><i class="fa fa-check"></i><b>26.1</b> scRNA-seq protocols</a></li>
<li class="chapter" data-level="26.2" data-path="resources.html"><a href="resources.html#external-rna-control-consortium-ercc"><i class="fa fa-check"></i><b>26.2</b> External RNA Control Consortium (ERCC)</a></li>
<li class="chapter" data-level="26.3" data-path="resources.html"><a href="resources.html#scrna-seq-analysis-tools"><i class="fa fa-check"></i><b>26.3</b> scRNA-seq analysis tools</a></li>
</ul></li>
<li class="chapter" data-level="27" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>27</b> References</a></li>
<li class="divider"></li>
<li><a href="http://www.sanger.ac.uk/science/groups/hemberg-group" target="blank">Hemberg Lab, 2016</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Analysis of single cell RNA-seq data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="clust-methods" class="section level1">
<h1><span class="header-section-number">19</span> Clustering example</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(scRNA.seq.funcs)
<span class="kw">library</span>(pcaMethods)
<span class="kw">library</span>(pcaReduce)
<span class="kw">library</span>(Rtsne)
<span class="kw">library</span>(SC3)
<span class="kw">library</span>(scater)
<span class="kw">library</span>(pheatmap)
<span class="kw">library</span>(ggplot2)
<span class="kw">set.seed</span>(<span class="dv">1234567</span>)</code></pre></div>
<p>To illustrate clustering of scRNA-seq data, we consider a dataset of hematopoietic stem cells (HSCs) collected from a patient with <a href="https://en.wikipedia.org/wiki/Myeloproliferative_neoplasm">myeloproliferative neoplasm (MPN)</a>. It is well known that this disease is highly heterogeneous with multiple sub-clones co-existing within the same patient.</p>
<div id="patient-dataset" class="section level2">
<h2><span class="header-section-number">19.1</span> Patient dataset</h2>
<p>Traditionally, clonal heterogeneity has been assessed by genotyping sub-clones. Genotyping through Sanger sequencing of two key loci has been carried out for this patient and it has revealed the presence of 3 different sub-clones (WT, WT/Tet2 and Jak2/Tet2). Our goal is to identify clusters corresponding to the three genotypes from the scRNA-seq data.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">patient.data &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;clustering/patient.txt&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>)</code></pre></div>
<p>For convenience we have performed all quality control and normalization steps (SF + RUVg) in advance. The dataset contains 51 cells and 8710 genes.</p>
<p>Now we are ready to cluster the data using the methods described in the previous chapter.</p>
</div>
<div id="sc3-1" class="section level2">
<h2><span class="header-section-number">19.2</span> SC3</h2>
<p><code>SC3</code> is a purely clustering tool and it does not provide functions for the sequencing quality control (QC) or normalisation. On the contrary it is expected that these preprocessing steps are performed by a user in advance. To encourage the preprocessing, SC3 is built on top of the <a href="http://bioconductor.org/packages/scater/">scater</a> package (see previous chapters). To our knowledge the scater is the most comprehensive toolkit for the QC and normalisation analysis of the single-cell RNA-Seq data.</p>
<p>If you already have an object of <code>scater</code> class <code>SCESet</code> then you should proceed to the next code chunk. If you have a matrix containing cell/gene expression values then you can create a <code>scater</code> object using your matrix:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">patient.sceset &lt;-<span class="st"> </span><span class="kw">newSCESet</span>(<span class="dt">countData =</span> patient.data)
patient.sceset &lt;-<span class="st"> </span><span class="kw">calculateQCMetrics</span>(patient.sceset)</code></pre></div>
<p>Now we are ready ro run the <code>SC3</code> clustering:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">patient.sceset &lt;-<span class="st"> </span><span class="kw">sc3</span>(patient.sceset, <span class="dt">ks =</span> <span class="dv">2</span>:<span class="dv">5</span>, <span class="dt">k_estimator =</span> <span class="ot">TRUE</span>, <span class="dt">biology =</span> <span class="ot">TRUE</span>)</code></pre></div>
<p>When the clustering is done you will be able to visualise the clustering results in many different ways. See the <a href="http://bioconductor.org/packages/release/bioc/vignettes/SC3/inst/doc/my-vignette.html">SC3 vignette</a> for more details.</p>
<p><code>SC3</code> also has a useful function for summarising/visualising the results in the interactive <code>Shiny</code> session:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sc3_interactive</span>(patient.sceset)</code></pre></div>
<p>This command will open <code>SC3</code> in a web browser. Once it is opened please perform the following exercises:</p>
<ul>
<li><p><strong>Exercise 1</strong>: Explore different clustering solutions for <span class="math inline">\(k\)</span> from 2 to 5. Also try to change the consensus averaging by checking and unchecking distance and transformation check boxes in the left panel of <strong>SC3</strong>.</p></li>
<li><p><strong>Exercise 2</strong>: Based on the genotyping and <code>SC3</code> prediction, we strongly believe that <span class="math inline">\(k =3\)</span> provides the best clustering. What evidence do you find for this in the clustering? How can you use the <a href="https://en.wikipedia.org/wiki/Silhouette_%28clustering%29">silhouette plots</a> to motivate choosing the value of <span class="math inline">\(k\)</span> that you think looks best?</p></li>
<li><p><strong>Exercise 3</strong>: Which clusters are the most stable? You can find out how different cells migrate between clusters using the “Cell Labels”/“Stability” tabs panel.</p></li>
<li><p><strong>Exercise 4</strong>: Check out differentially expressed genes and marker genes for the obtained clusterings. Please use <span class="math inline">\(k=3\)</span>.</p></li>
<li><p><strong>Exercise 5</strong>: Change the marker genes threshold (the default is 0.85). Does <strong>SC3</strong> find more marker genes?</p></li>
</ul>
</div>
<div id="pcareduce-1" class="section level2">
<h2><span class="header-section-number">19.3</span> pcaReduce</h2>
<p><code>pcaReduce</code> does not provide QC or normalisation function neither. It operates directly on the expression matrix. It is recommended to use a gene filter and log transformation before running <code>pcaReduce</code>. We will used the default <code>SC3</code> gene filter</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># use the same gene filter as in SC3</span>
input &lt;-<span class="st"> </span>patient.data[<span class="kw">fData</span>(patient.sceset)$sc3_gene_filter, ]
<span class="co"># log transform before the analysis</span>
input.log &lt;-<span class="st"> </span><span class="kw">log2</span>(input +<span class="st"> </span><span class="dv">1</span>)</code></pre></div>
<p>There are several parameters used by <code>pcaReduce</code>: * <code>nbt</code> defines a number of <code>pcaReduce</code> runs (it is stochastic and may have different solutions after different runs) * <code>q</code> defines number of dimensions to start clustering with. The output will contain partitions for all <span class="math inline">\(k\)</span> from 2 to q+1. * <code>method</code> defines a method used for clustering. <code>S</code> - to perform sampling based merging, <code>M</code> - to perform merging based on largest probability.</p>
<p>We will run <code>pcaReduce</code> 10 times:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># run pcaReduce 10 times creating hierarchies from 1 to 30 clusters</span>
pca.red &lt;-<span class="st"> </span><span class="kw">PCAreduce</span>(<span class="kw">t</span>(input.log), <span class="dt">nbt =</span> <span class="dv">10</span>, <span class="dt">q =</span> <span class="dv">30</span>, <span class="dt">method =</span> <span class="st">&#39;S&#39;</span>)</code></pre></div>
<p>Let’s compare the clusterings (with <span class="math inline">\(k\)</span> = 3) provided after running <code>pcaReduce</code> 10 times:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res &lt;-<span class="st"> </span><span class="kw">unique</span>(scRNA.seq.funcs::<span class="kw">merge_pcaReduce_results</span>(pca.red, <span class="dv">3</span>))
<span class="kw">pheatmap</span>(res, <span class="dt">legend =</span> <span class="ot">FALSE</span>, <span class="dt">show_rownames =</span> <span class="ot">FALSE</span>, <span class="dt">main =</span> <span class="st">&quot;Columns - cells, Rows - clusterings, Colors - clusters&quot;</span>)</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:clust-pca-reduce3"></span>
<img src="18-clustering_files/figure-html/clust-pca-reduce3-1.png" alt="Clustering solutions of pcaReduce method after running it for 10 times and selecting $k=3$" width="672" />
<p class="caption">
Figure 19.1: Clustering solutions of pcaReduce method after running it for 10 times and selecting <span class="math inline">\(k=3\)</span>
</p>
</div>
<p><strong>Exercise 6</strong>: Run pcaReduce for <span class="math inline">\(k=2\)</span>, <span class="math inline">\(k=4\)</span> and <span class="math inline">\(k=5\)</span>. Is it easy to choose the optimal <span class="math inline">\(k\)</span>?</p>
<p><strong>Hint</strong>: When running pcaReduce for different <span class="math inline">\(k\)</span>s you do not need to rerun pcaReduce::PCAreduce function, just use already calculated <code>pca.red</code> object.</p>
<strong>Our solutions</strong>:
<div class="figure" style="text-align: center"><span id="fig:clust-pca-reduce2"></span>
<img src="18-clustering_files/figure-html/clust-pca-reduce2-1.png" alt="Clustering solutions of pcaReduce method after running it for 10 times and selecting $k=2$." width="672" />
<p class="caption">
Figure 19.2: Clustering solutions of pcaReduce method after running it for 10 times and selecting <span class="math inline">\(k=2\)</span>.
</p>
</div>
<div class="figure" style="text-align: center"><span id="fig:clust-pca-reduce4"></span>
<img src="18-clustering_files/figure-html/clust-pca-reduce4-1.png" alt="Clustering solutions of pcaReduce method after running it for 10 times and selecting $k=4$." width="672" />
<p class="caption">
Figure 19.3: Clustering solutions of pcaReduce method after running it for 10 times and selecting <span class="math inline">\(k=4\)</span>.
</p>
</div>
<div class="figure" style="text-align: center"><span id="fig:clust-pca-reduce5"></span>
<img src="18-clustering_files/figure-html/clust-pca-reduce5-1.png" alt="Clustering solutions of pcaReduce method after running it for 10 times and selecting $k=5$." width="672" />
<p class="caption">
Figure 19.4: Clustering solutions of pcaReduce method after running it for 10 times and selecting <span class="math inline">\(k=5\)</span>.
</p>
</div>
<p><strong>Exercise 7</strong>: Compare the results between <code>SC3</code> and <code>pcaReduce</code>. What is the main difference between the solutions provided by the two different methods?</p>
</div>
<div id="tsne-kmeans" class="section level2">
<h2><span class="header-section-number">19.4</span> tSNE + kmeans</h2>
<p><a href="https://lvdmaaten.github.io/tsne/">tSNE</a> plots that we saw before (<a href="data-visualization.html#visual-tsne">10.3</a>) when used the <strong>scater</strong> package are made by using the <a href="https://cran.r-project.org/web/packages/Rtsne/index.html">Rtsne</a> and <a href="https://cran.r-project.org/web/packages/ggplot2/index.html">ggplot2</a> packages. We can create a similar plots explicitly:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tsne_out &lt;-<span class="st"> </span>Rtsne::<span class="kw">Rtsne</span>(<span class="kw">t</span>(input.log), <span class="dt">perplexity =</span> <span class="dv">10</span>)
df_to_plot &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(tsne_out$Y)
comps &lt;-<span class="st"> </span><span class="kw">colnames</span>(df_to_plot)[<span class="dv">1</span>:<span class="dv">2</span>]
ggplot2::<span class="kw">ggplot</span>(df_to_plot, <span class="kw">aes_string</span>(<span class="dt">x =</span> comps[<span class="dv">1</span>], <span class="dt">y =</span> comps[<span class="dv">2</span>])) +
<span class="st">    </span><span class="kw">geom_point</span>() +
<span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Dimension 1&quot;</span>) +
<span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Dimension 2&quot;</span>) +
<span class="st">    </span><span class="kw">theme_bw</span>()</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:clust-tsne"></span>
<img src="18-clustering_files/figure-html/clust-tsne-1.png" alt="tSNE map of the patient data" width="672" />
<p class="caption">
Figure 19.5: tSNE map of the patient data
</p>
</div>
<p>Note that all points on the plot above are black. This is different from what we saw before, when the cells were coloured based on the annotation. Here we do not have any annotation and all cells come from the same batch, therefore all dots are black.</p>
<p>Now we are going to apply <em>k</em>-means clustering algorithm to the cloud of points on the tSNE map. Do you see 2/3/4/5 groups in the cloud?</p>
<p>We will start with <span class="math inline">\(k=2\)</span>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">clusts &lt;-<span class="st"> </span><span class="kw">kmeans</span>(
    tsne_out$Y,
    <span class="dt">centers =</span> <span class="dv">2</span>,
    <span class="dt">iter.max =</span> <span class="fl">1e+09</span>,
    <span class="dt">nstart =</span> <span class="dv">1000</span>)$clust
df_to_plot$clusts &lt;-<span class="st"> </span><span class="kw">factor</span>(clusts, <span class="dt">levels =</span> <span class="kw">unique</span>(clusts))
comps &lt;-<span class="st"> </span><span class="kw">colnames</span>(df_to_plot)[<span class="dv">1</span>:<span class="dv">3</span>]
ggplot2::<span class="kw">ggplot</span>(df_to_plot,
                <span class="kw">aes_string</span>(<span class="dt">x =</span> comps[<span class="dv">1</span>], <span class="dt">y =</span> comps[<span class="dv">2</span>], <span class="dt">color =</span> comps[<span class="dv">3</span>])) +
<span class="st">    </span><span class="kw">geom_point</span>() +
<span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Dimension 1&quot;</span>) +
<span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Dimension 2&quot;</span>) +
<span class="st">    </span><span class="kw">theme_bw</span>()</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:clust-tsne-kmeans2"></span>
<img src="18-clustering_files/figure-html/clust-tsne-kmeans2-1.png" alt="tSNE map of the patient data with 2 colored clusters, identified by the k-means clustering algorithm" width="672" />
<p class="caption">
Figure 19.6: tSNE map of the patient data with 2 colored clusters, identified by the k-means clustering algorithm
</p>
</div>
<p><strong>Exercise 7</strong>: Make the same plots for <span class="math inline">\(k=3\)</span>, <span class="math inline">\(k=4\)</span> and <span class="math inline">\(k=5\)</span>.</p>
<p><strong>Exercise 8</strong>: Compare the results between <code>SC3</code> and <code>tSNE+kmeans</code>. Can the results be improved by changing the perplexity parameter?</p>
<p>As you may have noticed, both <code>pcaReduce</code> and <code>tSNE+kmeans</code> are stochastic and give different results every time they are run. To get a better overview of the solutions, we need to run the methods multiple times. Here we run <strong>tSNE+kmeans</strong> clustering 10 times with <span class="math inline">\(k = 3\)</span> (with perplexity = 10):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tsne.res &lt;-<span class="st"> </span>scRNA.seq.funcs::<span class="kw">tsne_mult</span>(input.log, <span class="dv">3</span>, <span class="dv">10</span>)
res &lt;-<span class="st"> </span><span class="kw">unique</span>(<span class="kw">do.call</span>(rbind, tsne.res))</code></pre></div>
<p><strong>Exercise 9</strong>: Visualize the different clustering solutions using a heatmap. Then run tSNE+kmeans algorithm with <span class="math inline">\(k = 2\)</span> or <span class="math inline">\(k = 4\)</span> and see how the clustering looks like in these cases.</p>
<div class="figure" style="text-align: center"><span id="fig:clust-tsne-kmeans3"></span>
<img src="18-clustering_files/figure-html/clust-tsne-kmeans3-1.png" alt="Clustering solutions of tSNE+kmeans method after running it for 10 times and using $k=3$" width="672" />
<p class="caption">
Figure 19.7: Clustering solutions of tSNE+kmeans method after running it for 10 times and using <span class="math inline">\(k=3\)</span>
</p>
</div>
<div class="figure" style="text-align: center"><span id="fig:clust-tsne-kmeans2-mult"></span>
<img src="18-clustering_files/figure-html/clust-tsne-kmeans2-mult-1.png" alt="Clustering solutions of tSNE+kmeans method after running it for 10 times and using $k=2$" width="672" />
<p class="caption">
Figure 19.8: Clustering solutions of tSNE+kmeans method after running it for 10 times and using <span class="math inline">\(k=2\)</span>
</p>
</div>
<div class="figure" style="text-align: center"><span id="fig:clust-tsne-kmeans4"></span>
<img src="18-clustering_files/figure-html/clust-tsne-kmeans4-1.png" alt="Clustering solutions of tSNE+kmeans method after running it for 10 times and using $k=4$" width="672" />
<p class="caption">
Figure 19.9: Clustering solutions of tSNE+kmeans method after running it for 10 times and using <span class="math inline">\(k=4\)</span>
</p>
</div>
</div>
<div id="snn-cliq-1" class="section level2">
<h2><span class="header-section-number">19.5</span> SNN-Cliq</h2>
<p>Here we run SNN-cliq with te default parameters provided in the author’s example:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">distan &lt;-<span class="st"> &quot;euclidean&quot;</span>
par.k &lt;-<span class="st"> </span><span class="dv">3</span>
par.r &lt;-<span class="st"> </span><span class="fl">0.7</span>
par.m &lt;-<span class="st"> </span><span class="fl">0.5</span>
<span class="co"># construct a graph</span>
scRNA.seq.funcs::<span class="kw">SNN</span>(
    <span class="dt">data =</span> <span class="kw">t</span>(input.log),
    <span class="dt">outfile =</span> <span class="st">&quot;snn-cliq.txt&quot;</span>,
    <span class="dt">k =</span> par.k,
    <span class="dt">distance =</span> distan
)
<span class="co"># find clusters in the graph</span>
snn.res &lt;-<span class="st"> </span>
<span class="st">    </span><span class="kw">system</span>(
        <span class="kw">paste0</span>(
            <span class="st">&quot;python snn-cliq/Cliq.py &quot;</span>, 
            <span class="st">&quot;-i snn-cliq.txt &quot;</span>,
            <span class="st">&quot;-o res-snn-cliq.txt &quot;</span>,
            <span class="st">&quot;-r &quot;</span>, par.r,
            <span class="st">&quot; -m &quot;</span>, par.m
        ),
        <span class="dt">intern =</span> <span class="ot">TRUE</span>
    )
<span class="kw">cat</span>(<span class="kw">paste</span>(snn.res, <span class="dt">collapse =</span> <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>))</code></pre></div>
<pre><code>## input file snn-cliq.txt
## find 7 quasi-cliques
## merged into 2 clusters
## unique assign done</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">snn.res &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;res-snn-cliq.txt&quot;</span>)
<span class="co"># remove files that were created during the analysis</span>
<span class="kw">system</span>(<span class="st">&quot;rm snn-cliq.txt res-snn-cliq.txt&quot;</span>)</code></pre></div>
<p><strong>Exercise 10</strong>: How can you characterize the solution identified by SNN-Cliq? Run SNN-Cliq algorithm with different values of <em>k</em>, <em>r</em>, <em>m</em> and <em>distance</em>, and see how the clustering looks like in these cases.</p>
</div>
<div id="sincera-1" class="section level2">
<h2><span class="header-section-number">19.6</span> SINCERA</h2>
<p>As mentioned in the previous chapter <a href="https://research.cchmc.org/pbge/sincera.html">SINCERA</a> is based on hierarchical clustering. One important thing to keep in mind is that it performs a gene-level z-score transformation before doing clustering:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># perform gene-by-gene per-sample z-score transformation</span>
dat &lt;-<span class="st"> </span><span class="kw">apply</span>(input, <span class="dv">1</span>, function(y) scRNA.seq.funcs::<span class="kw">z.transform.helper</span>(y))
<span class="co"># hierarchical clustering</span>
dd &lt;-<span class="st"> </span><span class="kw">as.dist</span>((<span class="dv">1</span> -<span class="st"> </span><span class="kw">cor</span>(<span class="kw">t</span>(dat), <span class="dt">method =</span> <span class="st">&quot;pearson&quot;</span>))/<span class="dv">2</span>)
hc &lt;-<span class="st"> </span><span class="kw">hclust</span>(dd, <span class="dt">method =</span> <span class="st">&quot;average&quot;</span>)</code></pre></div>
<p>If the number of cluster is not known <a href="https://research.cchmc.org/pbge/sincera.html">SINCERA</a> can identify <strong>k</strong> as the minimum height of the hierarchical tree that generates no more than a specified number of singleton clusters (clusters containing only 1 cell)</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">num.singleton &lt;-<span class="st"> </span><span class="dv">0</span>
kk &lt;-<span class="st"> </span><span class="dv">1</span>
for (i in <span class="dv">2</span>:<span class="kw">dim</span>(dat)[<span class="dv">2</span>]) {
    clusters &lt;-<span class="st"> </span><span class="kw">cutree</span>(hc, <span class="dt">k =</span> i)
    clustersizes &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">table</span>(clusters))
    singleton.clusters &lt;-<span class="st"> </span><span class="kw">which</span>(clustersizes$Freq &lt;<span class="st"> </span><span class="dv">2</span>)
    if (<span class="kw">length</span>(singleton.clusters) &lt;=<span class="st"> </span>num.singleton) {
        kk &lt;-<span class="st"> </span>i
    } else {
        break;
    }
}
<span class="kw">cat</span>(kk)</code></pre></div>
<pre><code>## 1</code></pre>
<p><strong>Exercise 11</strong>: Visualize the SINCERA results as a heatmap. How do the results compare to the other methods? What happens if you choose <span class="math inline">\(k = 3\)</span>?</p>
<strong>Our answer:</strong>
<div class="figure" style="text-align: center"><span id="fig:clust-sincera"></span>
<img src="18-clustering_files/figure-html/clust-sincera-1.png" alt="Clustering solutions of SINCERA method using $k=3$" width="672" />
<p class="caption">
Figure 19.10: Clustering solutions of SINCERA method using <span class="math inline">\(k=3\)</span>
</p>
</div>
<p><strong>Exercise 12</strong>: Is using the singleton cluster criteria for finding <strong>k</strong> a good idea?</p>
</div>
<div id="seurat-1" class="section level2">
<h2><span class="header-section-number">19.7</span> SEURAT</h2>
<p>Here we follow an <a href="http://satijalab.org/seurat/seurat_clustering_tutorial_part1.html">example</a> created by the authors of SEURAT. We had to introduce some modifications due to the errors produced by the original code:</p>
<p><strong>Note</strong> In the newest versions of SEURAT (v. 1.3-1.4) the tSNE is now used exclusively for visualization, and clustering is based on a <em>community detection</em> approach similar to one previously proposed for analyzing CyTOF data <span class="citation">(Levine et al. <a href="#ref-Levine2015-fk">2015</a>)</span>. In this tutorial we are still using an old version (v. 1.2) of <code>SEURAT</code> (this will be updated by the next time this course is run - approximately Spring 2017):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(Seurat)
<span class="co"># Create a SEURAT object</span>
d &lt;-<span class="st"> </span><span class="kw">new</span>(<span class="st">&quot;seurat&quot;</span>, <span class="dt">raw.data =</span> <span class="kw">as.data.frame</span>(<span class="kw">log</span>(patient.data +<span class="st"> </span><span class="dv">1</span>)))

<span class="co"># Setup a SEURAT object</span>
d &lt;-<span class="st"> </span><span class="kw">setup</span>(
    d,
    <span class="dt">project =</span> <span class="st">&quot;NBT&quot;</span>,
    <span class="dt">min.cells =</span> <span class="dv">3</span>,
    <span class="dt">names.field =</span> <span class="dv">2</span>,
    <span class="dt">names.delim =</span> <span class="st">&quot;_&quot;</span>,
    <span class="dt">min.genes =</span> <span class="dv">10</span>,
    <span class="dt">is.expr =</span> <span class="dv">1</span>
) 

<span class="co"># Genes placed into 20 bins based on X-axis (average expression).</span>
<span class="co"># Y-axis is within-bin z-score of log(Variance/mean). </span>
d &lt;-<span class="st"> </span><span class="kw">mean.var.plot</span>(
    d,
    <span class="dt">y.cutoff =</span> <span class="dv">2</span>,
    <span class="dt">x.low.cutoff =</span> <span class="dv">2</span>,
    <span class="dt">fxn.x =</span> expMean,
    <span class="dt">fxn.y =</span> logVarDivMean
)</code></pre></div>
<p><img src="18-clustering_files/figure-html/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Run a PCA using the variable genes as input</span>
d &lt;-<span class="st"> </span><span class="kw">pca</span>(d, <span class="dt">do.print =</span> <span class="ot">FALSE</span>, <span class="dt">pcs.store =</span> <span class="dv">25</span>)

<span class="co"># Do 200 random samplings to find significant genes, </span>
<span class="co"># each time randomly permute 1% of genes.</span>
<span class="co"># This returns a &#39;p-value&#39; for each gene in each PC, </span>
<span class="co"># based on how likely the gene/PC score would have been observed by chance</span>
d &lt;-<span class="st"> </span><span class="kw">jackStraw</span>(
    d,
    <span class="dt">num.replicate =</span> <span class="dv">200</span>,
    <span class="dt">do.print =</span> <span class="ot">FALSE</span>,
    <span class="dt">num.pc =</span> <span class="dv">25</span>
)

<span class="co"># Compare the distribution of P-values for each PC with a uniform distribution.</span>
<span class="co"># &#39;Significant&#39; PCs will have a strong enrichment of genes with low p-values</span>
pAll &lt;-<span class="st"> </span>d@jackStraw.empP
pAll$Contig &lt;-<span class="st"> </span><span class="kw">rownames</span>(pAll)
pAll.l &lt;-<span class="st"> </span><span class="kw">melt</span>(pAll, <span class="dt">id.vars =</span> <span class="st">&quot;Contig&quot;</span>)
<span class="kw">colnames</span>(pAll.l) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Contig&quot;</span>, <span class="st">&quot;PC&quot;</span>, <span class="st">&quot;Value&quot;</span>)

score.df &lt;-<span class="st"> </span><span class="ot">NULL</span>
score.thresh=<span class="fl">1e-5</span>
for (i in <span class="kw">unique</span>(pAll.l$PC)){
    q &lt;-<span class="st"> </span><span class="kw">qqplot</span>(pAll[, i], <span class="kw">runif</span>(<span class="dv">1000</span>), <span class="dt">plot.it =</span> <span class="ot">FALSE</span>)
    pc.score &lt;-<span class="st"> </span><span class="kw">prop.test</span>(
        <span class="kw">c</span>(
            <span class="kw">length</span>(<span class="kw">which</span>(pAll[, i] &lt;=<span class="st"> </span>score.thresh)), 
            <span class="kw">floor</span>(<span class="kw">nrow</span>(pAll) *<span class="st"> </span>score.thresh)
        ), 
        <span class="kw">c</span>(<span class="kw">nrow</span>(pAll), <span class="kw">nrow</span>(pAll))
    )$p.val
    if (<span class="kw">length</span>(<span class="kw">which</span>(pAll[, i] &lt;=<span class="st"> </span>score.thresh)) ==<span class="st"> </span><span class="dv">0</span>) pc.score &lt;-<span class="st"> </span><span class="dv">1</span>

    if(<span class="kw">is.null</span>(score.df))
      score.df &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">PC =</span> i, <span class="dt">Score =</span> pc.score)
    else
      score.df &lt;-<span class="st"> </span><span class="kw">rbind</span>(score.df, <span class="kw">data.frame</span>(<span class="dt">PC =</span> i, <span class="dt">Score =</span> pc.score))
}

<span class="co"># There are no significant PCs:</span>
<span class="kw">head</span>(score.df)</code></pre></div>
<pre><code>##    PC Score
## 1 PC1     1
## 2 PC2     1
## 3 PC3     1
## 4 PC4     1
## 5 PC5     1
## 6 PC6     1</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ndim &lt;-<span class="st"> </span><span class="dv">2</span>

<span class="co"># Project data on a 2D using tSNE</span>
d &lt;-<span class="st"> </span><span class="kw">run_tsne</span>(d, <span class="dt">dims.use =</span> <span class="dv">1</span>:ndim, <span class="dt">max_iter =</span> <span class="dv">2000</span>)

<span class="co"># Find clusters in the tSNE map</span>
d &lt;-<span class="st"> </span><span class="kw">DBclust_dimension</span>(
    d, 
    <span class="dv">1</span>, 
    <span class="dv">2</span>, 
    <span class="dt">reduction.use =</span> <span class="st">&quot;tsne&quot;</span>, 
    <span class="dt">G.use =</span> <span class="dv">8</span>, 
    <span class="dt">set.ident =</span> <span class="ot">TRUE</span>
)
<span class="kw">tsne.plot</span>(d, <span class="dt">pt.size =</span> <span class="dv">1</span>)</code></pre></div>
<p><img src="18-clustering_files/figure-html/unnamed-chunk-14-2.png" width="672" style="display: block; margin: auto;" /></p>
<p><strong>Exercise 13</strong>: As you can see DBSCAN could find only one cluster. It is known that DBSCAN cannot cluster data sets well with large differences in densities, since the <em>G.use</em> parameter cannot then be chosen appropriately for all clusters. Try to change <em>G.use</em> to be able to find more than one cluster in the data.</p>
<p><strong>Note</strong> We found that in general SEURAT does not work well for small datasets (<em>N</em> &lt; 200 cells). For a comprehensive comparisons of the methods please look at the SC3 paper <span class="citation">(Kiselev et al. <a href="#ref-Kiselev2016-bq">2016</a>)</span>.</p>

</div>
</div>
<h3> References</h3>
<div id="refs" class="references">
<div id="ref-Levine2015-fk">
<p>Levine, Jacob H, Erin F Simonds, Sean C Bendall, Kara L Davis, El-Ad D Amir, Michelle D Tadmor, Oren Litvin, et al. 2015. “Data-Driven Phenotypic Dissection of AML Reveals Progenitor-Like Cells That Correlate with Prognosis.” <em>Cell</em> 162 (1): 184–97.</p>
</div>
<div id="ref-Kiselev2016-bq">
<p>Kiselev, Vladimir Yu, Kristina Kirschner, Michael T Schaub, Tallulah Andrews, Tamir Chandra, Kedar N Natarajan, Wolf Reik, Mauricio Barahona, Anthony R Green, and Martin Hemberg. 2016. “SC3 - Consensus Clustering of Single-Cell RNA-Seq Data.” <em>BioRxiv</em>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="clustering-introduction.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="identification-of-important-genes.html" class="navigation navigation-next " aria-label="Next page""><i class="fa fa-angle-right"></i></a>

<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/hemberg-lab/scRNA.seq.course/edit/master/18-clustering.Rmd",
"text": "Edit"
},
"download": ["scRNA-seq-course.pdf"],
"toc": {
"collapse": "section"
},
"search": true
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
