<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Analysis of single cell RNA-seq data</title>
  <meta content="text/html; charset=UTF-8" http-equiv="Content-Type">
  <meta name="description" content="Analysis of single cell RNA-seq data">
  <meta name="generator" content="bookdown 0.3 and GitBook 2.6.7">

  <meta property="og:title" content="Analysis of single cell RNA-seq data" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Analysis of single cell RNA-seq data" />
  
  
  

<meta name="author" content="Vladimir Kiselev, Tallulah Andrews, Davis McCarthy and Martin Hemberg">


<meta name="date" content="2017-03-12">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="dealing-with-confounders-reads.html">
<link rel="next" href="clust-methods.html">

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<!-- for Facebook -->  
<meta property="og:url" content="http://hemberg-lab.github.io/scRNA.seq.course/" />
<meta property="og:description" content="In this course we will be surveying the existing problems as well as the available computational and statistical frameworks available for the analysis of scRNA-seq. The course is taught through the University of Cambridge Bioinformatics training unit, but the material found on these pages is meant to be used for anyone interested in learning about computational analysis of scRNA-seq data." />
<meta property="og:image" content="http://hemberg-lab.github.io/scRNA.seq.course/figures/RNA-Seq_workflow-5.pdf.jpg" />

<!-- for Twitter -->          
<meta name="twitter:card" content="summary_large_image" />
<meta name="twitter:title" content="Analysis of single-cell RNA-seq data" />
<meta name="twitter:description" content="In this course we will be surveying the existing problems as well as the available computational and statistical frameworks available for the analysis of scRNA-seq. The course is taught through the University of Cambridge Bioinformatics training unit, but the material found on these pages is meant to be used for anyone interested in learning about computational analysis of scRNA-seq data." />
<meta name="twitter:image" content="http://hemberg-lab.github.io/scRNA.seq.course/figures/RNA-Seq_workflow-5.pdf.jpg" />

<!-- Google Analytics -->
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-71525309-1', 'auto');
  ga('send', 'pageview');

</script>


<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>


  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="index.html">Table of Contents</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About the course</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#video"><i class="fa fa-check"></i><b>1.1</b> Video</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#registration"><i class="fa fa-check"></i><b>1.2</b> Registration</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#github"><i class="fa fa-check"></i><b>1.3</b> GitHub</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#docker-container"><i class="fa fa-check"></i><b>1.4</b> Docker container</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>1.5</b> License</a></li>
<li class="chapter" data-level="1.6" data-path="index.html"><a href="index.html#prerequisites"><i class="fa fa-check"></i><b>1.6</b> Prerequisites</a></li>
<li class="chapter" data-level="1.7" data-path="index.html"><a href="index.html#contact"><i class="fa fa-check"></i><b>1.7</b> Contact</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="technical-requirements.html"><a href="technical-requirements.html"><i class="fa fa-check"></i><b>2</b> Technical requirements</a><ul>
<li class="chapter" data-level="2.1" data-path="technical-requirements.html"><a href="technical-requirements.html#general"><i class="fa fa-check"></i><b>2.1</b> General</a></li>
<li class="chapter" data-level="2.2" data-path="technical-requirements.html"><a href="technical-requirements.html#plotting"><i class="fa fa-check"></i><b>2.2</b> Plotting</a></li>
<li class="chapter" data-level="2.3" data-path="technical-requirements.html"><a href="technical-requirements.html#qc-and-normalisation"><i class="fa fa-check"></i><b>2.3</b> QC and normalisation</a></li>
<li class="chapter" data-level="2.4" data-path="technical-requirements.html"><a href="technical-requirements.html#clustering"><i class="fa fa-check"></i><b>2.4</b> Clustering</a></li>
<li class="chapter" data-level="2.5" data-path="technical-requirements.html"><a href="technical-requirements.html#dropouts"><i class="fa fa-check"></i><b>2.5</b> Dropouts</a></li>
<li class="chapter" data-level="2.6" data-path="technical-requirements.html"><a href="technical-requirements.html#pseudotime"><i class="fa fa-check"></i><b>2.6</b> Pseudotime</a></li>
<li class="chapter" data-level="2.7" data-path="technical-requirements.html"><a href="technical-requirements.html#differential-expression"><i class="fa fa-check"></i><b>2.7</b> Differential Expression</a></li>
<li class="chapter" data-level="2.8" data-path="technical-requirements.html"><a href="technical-requirements.html#extra-tools"><i class="fa fa-check"></i><b>2.8</b> Extra tools</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html"><i class="fa fa-check"></i><b>3</b> Introduction to single-cell RNA-seq</a><ul>
<li class="chapter" data-level="3.1" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#bulk-rna-seq"><i class="fa fa-check"></i><b>3.1</b> Bulk RNA-seq</a></li>
<li class="chapter" data-level="3.2" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#scrna-seq"><i class="fa fa-check"></i><b>3.2</b> scRNA-seq</a></li>
<li class="chapter" data-level="3.3" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#workflow"><i class="fa fa-check"></i><b>3.3</b> Workflow</a></li>
<li class="chapter" data-level="3.4" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#computational-analysis"><i class="fa fa-check"></i><b>3.4</b> Computational Analysis</a></li>
<li class="chapter" data-level="3.5" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#challenges"><i class="fa fa-check"></i><b>3.5</b> Challenges</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="overview-of-experimental-methods-for-generating-scrna-seq-data.html"><a href="overview-of-experimental-methods-for-generating-scrna-seq-data.html"><i class="fa fa-check"></i><b>4</b> Overview of experimental methods for generating scRNA-seq data</a></li>
<li class="chapter" data-level="5" data-path="what-platform-to-use-for-my-experiment.html"><a href="what-platform-to-use-for-my-experiment.html"><i class="fa fa-check"></i><b>5</b> What platform to use for my experiment?</a></li>
<li class="chapter" data-level="6" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html"><i class="fa fa-check"></i><b>6</b> Construction of expression matrix</a><ul>
<li class="chapter" data-level="6.1" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#reads-qc"><i class="fa fa-check"></i><b>6.1</b> Reads QC</a></li>
<li class="chapter" data-level="6.2" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#reads-alignment"><i class="fa fa-check"></i><b>6.2</b> Reads alignment</a></li>
<li class="chapter" data-level="6.3" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#alignment-example"><i class="fa fa-check"></i><b>6.3</b> Alignment example</a></li>
<li class="chapter" data-level="6.4" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#mapping-qc"><i class="fa fa-check"></i><b>6.4</b> Mapping QC</a></li>
<li class="chapter" data-level="6.5" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#reads-quantification"><i class="fa fa-check"></i><b>6.5</b> Reads quantification</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="unique-molecular-identifiers-umis.html"><a href="unique-molecular-identifiers-umis.html"><i class="fa fa-check"></i><b>7</b> Unique Molecular Identifiers (UMIs)</a><ul>
<li class="chapter" data-level="7.1" data-path="unique-molecular-identifiers-umis.html"><a href="unique-molecular-identifiers-umis.html#introduction"><i class="fa fa-check"></i><b>7.1</b> Introduction</a></li>
<li class="chapter" data-level="7.2" data-path="unique-molecular-identifiers-umis.html"><a href="unique-molecular-identifiers-umis.html#mapping-barcodes"><i class="fa fa-check"></i><b>7.2</b> Mapping Barcodes</a></li>
<li class="chapter" data-level="7.3" data-path="unique-molecular-identifiers-umis.html"><a href="unique-molecular-identifiers-umis.html#counting-barcodes"><i class="fa fa-check"></i><b>7.3</b> Counting Barcodes</a></li>
<li class="chapter" data-level="7.4" data-path="unique-molecular-identifiers-umis.html"><a href="unique-molecular-identifiers-umis.html#correcting-for-errors"><i class="fa fa-check"></i><b>7.4</b> Correcting for Errors</a></li>
<li class="chapter" data-level="7.5" data-path="unique-molecular-identifiers-umis.html"><a href="unique-molecular-identifiers-umis.html#downstream-analysis"><i class="fa fa-check"></i><b>7.5</b> Downstream Analysis</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="scater-package.html"><a href="scater-package.html"><i class="fa fa-check"></i><b>8</b> scater package</a><ul>
<li class="chapter" data-level="8.1" data-path="scater-package.html"><a href="scater-package.html#introduction-1"><i class="fa fa-check"></i><b>8.1</b> Introduction</a></li>
<li class="chapter" data-level="8.2" data-path="scater-package.html"><a href="scater-package.html#scater-workflow"><i class="fa fa-check"></i><b>8.2</b> scater workflow</a></li>
<li class="chapter" data-level="8.3" data-path="scater-package.html"><a href="scater-package.html#terminology"><i class="fa fa-check"></i><b>8.3</b> Terminology</a></li>
<li class="chapter" data-level="8.4" data-path="scater-package.html"><a href="scater-package.html#sceset-class"><i class="fa fa-check"></i><b>8.4</b> SCESet class</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html"><i class="fa fa-check"></i><b>9</b> Expression QC (UMI)</a><ul>
<li class="chapter" data-level="9.1" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#introduction-2"><i class="fa fa-check"></i><b>9.1</b> Introduction</a></li>
<li class="chapter" data-level="9.2" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#blischak-dataset"><i class="fa fa-check"></i><b>9.2</b> Blischak dataset</a></li>
<li class="chapter" data-level="9.3" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#cell-qc"><i class="fa fa-check"></i><b>9.3</b> Cell QC</a><ul>
<li class="chapter" data-level="9.3.1" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#library-size"><i class="fa fa-check"></i><b>9.3.1</b> Library size</a></li>
<li class="chapter" data-level="9.3.2" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#detected-genes-1"><i class="fa fa-check"></i><b>9.3.2</b> Detected genes (1)</a></li>
<li class="chapter" data-level="9.3.3" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#erccs-and-mts"><i class="fa fa-check"></i><b>9.3.3</b> ERCCs and MTs</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#cell-filtering"><i class="fa fa-check"></i><b>9.4</b> Cell filtering</a><ul>
<li class="chapter" data-level="9.4.1" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#manual"><i class="fa fa-check"></i><b>9.4.1</b> Manual</a></li>
<li class="chapter" data-level="9.4.2" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#default-thresholds"><i class="fa fa-check"></i><b>9.4.2</b> Default Thresholds</a></li>
<li class="chapter" data-level="9.4.3" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#automatic"><i class="fa fa-check"></i><b>9.4.3</b> Automatic</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#compare-filterings"><i class="fa fa-check"></i><b>9.5</b> Compare filterings</a></li>
<li class="chapter" data-level="9.6" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#gene-analysis"><i class="fa fa-check"></i><b>9.6</b> Gene analysis</a><ul>
<li class="chapter" data-level="9.6.1" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#gene-expression"><i class="fa fa-check"></i><b>9.6.1</b> Gene expression</a></li>
<li class="chapter" data-level="9.6.2" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#gene-filtering"><i class="fa fa-check"></i><b>9.6.2</b> Gene filtering</a></li>
</ul></li>
<li class="chapter" data-level="9.7" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#save-the-data"><i class="fa fa-check"></i><b>9.7</b> Save the data</a></li>
<li class="chapter" data-level="9.8" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#big-exercise"><i class="fa fa-check"></i><b>9.8</b> Big Exercise</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="expression-qc-reads.html"><a href="expression-qc-reads.html"><i class="fa fa-check"></i><b>10</b> Expression QC (Reads)</a></li>
<li class="chapter" data-level="11" data-path="data-visualization.html"><a href="data-visualization.html"><i class="fa fa-check"></i><b>11</b> Data visualization</a><ul>
<li class="chapter" data-level="11.1" data-path="data-visualization.html"><a href="data-visualization.html#introduction-3"><i class="fa fa-check"></i><b>11.1</b> Introduction</a></li>
<li class="chapter" data-level="11.2" data-path="data-visualization.html"><a href="data-visualization.html#visual-pca"><i class="fa fa-check"></i><b>11.2</b> PCA plot</a><ul>
<li class="chapter" data-level="11.2.1" data-path="data-visualization.html"><a href="data-visualization.html#before-qc"><i class="fa fa-check"></i><b>11.2.1</b> Before QC</a></li>
<li class="chapter" data-level="11.2.2" data-path="data-visualization.html"><a href="data-visualization.html#after-qc"><i class="fa fa-check"></i><b>11.2.2</b> After QC</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="data-visualization.html"><a href="data-visualization.html#visual-tsne"><i class="fa fa-check"></i><b>11.3</b> tSNE map</a><ul>
<li class="chapter" data-level="11.3.1" data-path="data-visualization.html"><a href="data-visualization.html#before-qc-1"><i class="fa fa-check"></i><b>11.3.1</b> Before QC</a></li>
<li class="chapter" data-level="11.3.2" data-path="data-visualization.html"><a href="data-visualization.html#after-qc-1"><i class="fa fa-check"></i><b>11.3.2</b> After QC</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="data-visualization.html"><a href="data-visualization.html#big-exercise-1"><i class="fa fa-check"></i><b>11.4</b> Big Exercise</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="data-visualization-reads.html"><a href="data-visualization-reads.html"><i class="fa fa-check"></i><b>12</b> Data visualization (Reads)</a></li>
<li class="chapter" data-level="13" data-path="confounding-factors.html"><a href="confounding-factors.html"><i class="fa fa-check"></i><b>13</b> Confounding factors</a><ul>
<li class="chapter" data-level="13.1" data-path="confounding-factors.html"><a href="confounding-factors.html#introduction-4"><i class="fa fa-check"></i><b>13.1</b> Introduction</a></li>
<li class="chapter" data-level="13.2" data-path="confounding-factors.html"><a href="confounding-factors.html#correlations-with-pcs"><i class="fa fa-check"></i><b>13.2</b> Correlations with PCs</a><ul>
<li class="chapter" data-level="13.2.1" data-path="confounding-factors.html"><a href="confounding-factors.html#detected-genes"><i class="fa fa-check"></i><b>13.2.1</b> Detected genes</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="confounding-factors.html"><a href="confounding-factors.html#explanatory-variables"><i class="fa fa-check"></i><b>13.3</b> Explanatory variables</a></li>
<li class="chapter" data-level="13.4" data-path="confounding-factors.html"><a href="confounding-factors.html#other-confounders"><i class="fa fa-check"></i><b>13.4</b> Other confounders</a></li>
<li class="chapter" data-level="13.5" data-path="confounding-factors.html"><a href="confounding-factors.html#exercise"><i class="fa fa-check"></i><b>13.5</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="confounding-factors-reads.html"><a href="confounding-factors-reads.html"><i class="fa fa-check"></i><b>14</b> Confounding factors (Reads)</a></li>
<li class="chapter" data-level="15" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html"><i class="fa fa-check"></i><b>15</b> Normalization for library size</a><ul>
<li class="chapter" data-level="15.1" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#introduction-5"><i class="fa fa-check"></i><b>15.1</b> Introduction</a></li>
<li class="chapter" data-level="15.2" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#library-size-1"><i class="fa fa-check"></i><b>15.2</b> Library size</a></li>
<li class="chapter" data-level="15.3" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#normalisations"><i class="fa fa-check"></i><b>15.3</b> Normalisations</a><ul>
<li class="chapter" data-level="15.3.1" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#raw"><i class="fa fa-check"></i><b>15.3.1</b> Raw</a></li>
<li class="chapter" data-level="15.3.2" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#cpm"><i class="fa fa-check"></i><b>15.3.2</b> CPM</a></li>
<li class="chapter" data-level="15.3.3" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#tmm"><i class="fa fa-check"></i><b>15.3.3</b> TMM</a></li>
<li class="chapter" data-level="15.3.4" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#scran"><i class="fa fa-check"></i><b>15.3.4</b> scran</a></li>
<li class="chapter" data-level="15.3.5" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#size-factor-rle"><i class="fa fa-check"></i><b>15.3.5</b> Size-factor (RLE)</a></li>
<li class="chapter" data-level="15.3.6" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#upperquantile"><i class="fa fa-check"></i><b>15.3.6</b> Upperquantile</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#downsampling"><i class="fa fa-check"></i><b>15.4</b> Downsampling</a></li>
<li class="chapter" data-level="15.5" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#normalizing-for-genetranscript-length"><i class="fa fa-check"></i><b>15.5</b> Normalizing for gene/transcript length</a></li>
<li class="chapter" data-level="15.6" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#exercise-1"><i class="fa fa-check"></i><b>15.6</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="normalization-for-library-size-reads.html"><a href="normalization-for-library-size-reads.html"><i class="fa fa-check"></i><b>16</b> Normalization for library size (Reads)</a></li>
<li class="chapter" data-level="17" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html"><i class="fa fa-check"></i><b>17</b> Dealing with confounders</a><ul>
<li class="chapter" data-level="17.1" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html#introduction-6"><i class="fa fa-check"></i><b>17.1</b> Introduction</a></li>
<li class="chapter" data-level="17.2" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html#remove-unwanted-variation"><i class="fa fa-check"></i><b>17.2</b> Remove Unwanted Variation</a><ul>
<li class="chapter" data-level="17.2.1" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html#ruvg"><i class="fa fa-check"></i><b>17.2.1</b> RUVg</a></li>
<li class="chapter" data-level="17.2.2" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html#ruvs"><i class="fa fa-check"></i><b>17.2.2</b> RUVs</a></li>
</ul></li>
<li class="chapter" data-level="17.3" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html#effectiveness-1"><i class="fa fa-check"></i><b>17.3</b> Effectiveness 1</a></li>
<li class="chapter" data-level="17.4" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html#effectiveness-2"><i class="fa fa-check"></i><b>17.4</b> Effectiveness 2</a></li>
<li class="chapter" data-level="17.5" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html#effectiveness-3"><i class="fa fa-check"></i><b>17.5</b> Effectiveness 3</a><ul>
<li class="chapter" data-level="17.5.1" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html#de-raw-counts"><i class="fa fa-check"></i><b>17.5.1</b> DE (raw counts)</a></li>
<li class="chapter" data-level="17.5.2" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html#de-ruvg-k-2"><i class="fa fa-check"></i><b>17.5.2</b> DE (RUVg, k = 2)</a></li>
<li class="chapter" data-level="17.5.3" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html#de-ruvs-k-2"><i class="fa fa-check"></i><b>17.5.3</b> DE (RUVs, k = 2)</a></li>
</ul></li>
<li class="chapter" data-level="17.6" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html#exercise-2"><i class="fa fa-check"></i><b>17.6</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="dealing-with-confounders-reads.html"><a href="dealing-with-confounders-reads.html"><i class="fa fa-check"></i><b>18</b> Dealing with confounders (Reads)</a><ul>
<li class="chapter" data-level="18.1" data-path="dealing-with-confounders-reads.html"><a href="dealing-with-confounders-reads.html#remove-unwanted-variation-1"><i class="fa fa-check"></i><b>18.1</b> Remove Unwanted Variation</a><ul>
<li class="chapter" data-level="18.1.1" data-path="dealing-with-confounders-reads.html"><a href="dealing-with-confounders-reads.html#ruvg-1"><i class="fa fa-check"></i><b>18.1.1</b> RUVg</a></li>
<li class="chapter" data-level="18.1.2" data-path="dealing-with-confounders-reads.html"><a href="dealing-with-confounders-reads.html#ruvs-1"><i class="fa fa-check"></i><b>18.1.2</b> RUVs</a></li>
</ul></li>
<li class="chapter" data-level="18.2" data-path="dealing-with-confounders-reads.html"><a href="dealing-with-confounders-reads.html#effectiveness-1-1"><i class="fa fa-check"></i><b>18.2</b> Effectiveness 1</a></li>
<li class="chapter" data-level="18.3" data-path="dealing-with-confounders-reads.html"><a href="dealing-with-confounders-reads.html#effectiveness-2-1"><i class="fa fa-check"></i><b>18.3</b> Effectiveness 2</a></li>
<li class="chapter" data-level="18.4" data-path="dealing-with-confounders-reads.html"><a href="dealing-with-confounders-reads.html#effectiveness-3-1"><i class="fa fa-check"></i><b>18.4</b> Effectiveness 3</a><ul>
<li class="chapter" data-level="18.4.1" data-path="dealing-with-confounders-reads.html"><a href="dealing-with-confounders-reads.html#de-raw-counts-1"><i class="fa fa-check"></i><b>18.4.1</b> DE (raw counts)</a></li>
<li class="chapter" data-level="18.4.2" data-path="dealing-with-confounders-reads.html"><a href="dealing-with-confounders-reads.html#de-ruvg-k-2-1"><i class="fa fa-check"></i><b>18.4.2</b> DE (RUVg, k = 2)</a></li>
<li class="chapter" data-level="18.4.3" data-path="dealing-with-confounders-reads.html"><a href="dealing-with-confounders-reads.html#de-ruvs-k-2-1"><i class="fa fa-check"></i><b>18.4.3</b> DE (RUVs, k = 2)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="19" data-path="clustering-introduction.html"><a href="clustering-introduction.html"><i class="fa fa-check"></i><b>19</b> Clustering Introduction</a><ul>
<li class="chapter" data-level="19.1" data-path="clustering-introduction.html"><a href="clustering-introduction.html#introduction-7"><i class="fa fa-check"></i><b>19.1</b> Introduction</a></li>
<li class="chapter" data-level="19.2" data-path="clustering-introduction.html"><a href="clustering-introduction.html#dimensionality-reductions"><i class="fa fa-check"></i><b>19.2</b> Dimensionality reductions</a><ul>
<li class="chapter" data-level="19.2.1" data-path="clustering-introduction.html"><a href="clustering-introduction.html#pca"><i class="fa fa-check"></i><b>19.2.1</b> PCA</a></li>
<li class="chapter" data-level="19.2.2" data-path="clustering-introduction.html"><a href="clustering-introduction.html#spectral"><i class="fa fa-check"></i><b>19.2.2</b> Spectral</a></li>
<li class="chapter" data-level="19.2.3" data-path="clustering-introduction.html"><a href="clustering-introduction.html#tsne"><i class="fa fa-check"></i><b>19.2.3</b> tSNE</a></li>
</ul></li>
<li class="chapter" data-level="19.3" data-path="clustering-introduction.html"><a href="clustering-introduction.html#clustering-methods"><i class="fa fa-check"></i><b>19.3</b> Clustering methods</a><ul>
<li class="chapter" data-level="19.3.1" data-path="clustering-introduction.html"><a href="clustering-introduction.html#hierarchical-clustering"><i class="fa fa-check"></i><b>19.3.1</b> Hierarchical clustering</a></li>
<li class="chapter" data-level="19.3.2" data-path="clustering-introduction.html"><a href="clustering-introduction.html#k-means"><i class="fa fa-check"></i><b>19.3.2</b> k-means</a></li>
<li class="chapter" data-level="19.3.3" data-path="clustering-introduction.html"><a href="clustering-introduction.html#graph-based-methods"><i class="fa fa-check"></i><b>19.3.3</b> Graph-based methods</a></li>
</ul></li>
<li class="chapter" data-level="19.4" data-path="clustering-introduction.html"><a href="clustering-introduction.html#challenges-in-clustering"><i class="fa fa-check"></i><b>19.4</b> Challenges in clustering</a></li>
<li class="chapter" data-level="19.5" data-path="clustering-introduction.html"><a href="clustering-introduction.html#tools-for-scrna-seq-data"><i class="fa fa-check"></i><b>19.5</b> Tools for scRNA-seq data</a><ul>
<li class="chapter" data-level="19.5.1" data-path="clustering-introduction.html"><a href="clustering-introduction.html#sincera"><i class="fa fa-check"></i><b>19.5.1</b> <a href="https://research.cchmc.org/pbge/sincera.html">SINCERA</a></a></li>
<li class="chapter" data-level="19.5.2" data-path="clustering-introduction.html"><a href="clustering-introduction.html#pcareduce"><i class="fa fa-check"></i><b>19.5.2</b> <a href="https://github.com/JustinaZ/pcaReduce">pcaReduce</a></a></li>
<li class="chapter" data-level="19.5.3" data-path="clustering-introduction.html"><a href="clustering-introduction.html#sc3"><i class="fa fa-check"></i><b>19.5.3</b> <a href="http://bioconductor.org/packages/SC3/">SC3</a></a></li>
<li class="chapter" data-level="19.5.4" data-path="clustering-introduction.html"><a href="clustering-introduction.html#tsne-k-means"><i class="fa fa-check"></i><b>19.5.4</b> tSNE + k-means</a></li>
<li class="chapter" data-level="19.5.5" data-path="clustering-introduction.html"><a href="clustering-introduction.html#seurat"><i class="fa fa-check"></i><b>19.5.5</b> <a href="https://github.com/Puriney/seurat">SEURAT</a></a></li>
<li class="chapter" data-level="19.5.6" data-path="clustering-introduction.html"><a href="clustering-introduction.html#snn-cliq"><i class="fa fa-check"></i><b>19.5.6</b> <a href="http://bioinfo.uncc.edu/SNNCliq/">SNN-Cliq</a></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="20" data-path="clust-methods.html"><a href="clust-methods.html"><i class="fa fa-check"></i><b>20</b> Clustering example</a><ul>
<li class="chapter" data-level="20.1" data-path="clust-methods.html"><a href="clust-methods.html#patient-dataset"><i class="fa fa-check"></i><b>20.1</b> Patient dataset</a></li>
<li class="chapter" data-level="20.2" data-path="clust-methods.html"><a href="clust-methods.html#sc3-1"><i class="fa fa-check"></i><b>20.2</b> SC3</a></li>
<li class="chapter" data-level="20.3" data-path="clust-methods.html"><a href="clust-methods.html#pcareduce-1"><i class="fa fa-check"></i><b>20.3</b> pcaReduce</a></li>
<li class="chapter" data-level="20.4" data-path="clust-methods.html"><a href="clust-methods.html#tsne-kmeans"><i class="fa fa-check"></i><b>20.4</b> tSNE + kmeans</a></li>
<li class="chapter" data-level="20.5" data-path="clust-methods.html"><a href="clust-methods.html#snn-cliq-1"><i class="fa fa-check"></i><b>20.5</b> SNN-Cliq</a></li>
<li class="chapter" data-level="20.6" data-path="clust-methods.html"><a href="clust-methods.html#sincera-1"><i class="fa fa-check"></i><b>20.6</b> SINCERA</a></li>
<li class="chapter" data-level="20.7" data-path="clust-methods.html"><a href="clust-methods.html#seurat-1"><i class="fa fa-check"></i><b>20.7</b> SEURAT</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="identification-of-important-genes.html"><a href="identification-of-important-genes.html"><i class="fa fa-check"></i><b>21</b> Identification of important genes</a><ul>
<li class="chapter" data-level="21.1" data-path="identification-of-important-genes.html"><a href="identification-of-important-genes.html#fitting-the-models"><i class="fa fa-check"></i><b>21.1</b> Fitting the models</a></li>
<li class="chapter" data-level="21.2" data-path="identification-of-important-genes.html"><a href="identification-of-important-genes.html#the-michaelis-menten-equation"><i class="fa fa-check"></i><b>21.2</b> The Michaelis-Menten Equation</a></li>
<li class="chapter" data-level="21.3" data-path="identification-of-important-genes.html"><a href="identification-of-important-genes.html#right-outliers"><i class="fa fa-check"></i><b>21.3</b> Right outliers</a></li>
<li class="chapter" data-level="21.4" data-path="identification-of-important-genes.html"><a href="identification-of-important-genes.html#validation-of-de-results"><i class="fa fa-check"></i><b>21.4</b> Validation of DE results</a></li>
<li class="chapter" data-level="21.5" data-path="identification-of-important-genes.html"><a href="identification-of-important-genes.html#comparing-m3drop-to-other-methods"><i class="fa fa-check"></i><b>21.5</b> Comparing M3Drop to other methods</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="pseudotime-analysis.html"><a href="pseudotime-analysis.html"><i class="fa fa-check"></i><b>22</b> Pseudotime analysis</a><ul>
<li class="chapter" data-level="22.1" data-path="pseudotime-analysis.html"><a href="pseudotime-analysis.html#tscan"><i class="fa fa-check"></i><b>22.1</b> TSCAN</a></li>
<li class="chapter" data-level="22.2" data-path="pseudotime-analysis.html"><a href="pseudotime-analysis.html#monocle"><i class="fa fa-check"></i><b>22.2</b> monocle</a></li>
<li class="chapter" data-level="22.3" data-path="pseudotime-analysis.html"><a href="pseudotime-analysis.html#diffusion-maps"><i class="fa fa-check"></i><b>22.3</b> Diffusion maps</a></li>
<li class="chapter" data-level="22.4" data-path="pseudotime-analysis.html"><a href="pseudotime-analysis.html#slicer"><i class="fa fa-check"></i><b>22.4</b> SLICER</a></li>
<li class="chapter" data-level="22.5" data-path="pseudotime-analysis.html"><a href="pseudotime-analysis.html#comparison-of-the-methods"><i class="fa fa-check"></i><b>22.5</b> Comparison of the methods</a></li>
<li class="chapter" data-level="22.6" data-path="pseudotime-analysis.html"><a href="pseudotime-analysis.html#expression-of-genes-through-time"><i class="fa fa-check"></i><b>22.6</b> Expression of genes through time</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="differential-expression-de-analysis.html"><a href="differential-expression-de-analysis.html"><i class="fa fa-check"></i><b>23</b> Differential Expression (DE) analysis</a><ul>
<li class="chapter" data-level="23.1" data-path="differential-expression-de-analysis.html"><a href="differential-expression-de-analysis.html#bulk-rna-seq-1"><i class="fa fa-check"></i><b>23.1</b> Bulk RNA-seq</a></li>
<li class="chapter" data-level="23.2" data-path="differential-expression-de-analysis.html"><a href="differential-expression-de-analysis.html#single-cell-rna-seq"><i class="fa fa-check"></i><b>23.2</b> Single cell RNA-seq</a></li>
<li class="chapter" data-level="23.3" data-path="differential-expression-de-analysis.html"><a href="differential-expression-de-analysis.html#differences-in-distribution"><i class="fa fa-check"></i><b>23.3</b> Differences in Distribution</a></li>
<li class="chapter" data-level="23.4" data-path="differential-expression-de-analysis.html"><a href="differential-expression-de-analysis.html#models-of-single-cell-rnaseq-data"><i class="fa fa-check"></i><b>23.4</b> Models of single-cell RNASeq data</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html"><i class="fa fa-check"></i><b>24</b> DE in a real dataset</a><ul>
<li class="chapter" data-level="24.1" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#introduction-8"><i class="fa fa-check"></i><b>24.1</b> Introduction</a></li>
<li class="chapter" data-level="24.2" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#kolmogorov-smirnov-test"><i class="fa fa-check"></i><b>24.2</b> Kolmogorov-Smirnov test</a><ul>
<li class="chapter" data-level="24.2.1" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#evaluating-accuracy"><i class="fa fa-check"></i><b>24.2.1</b> Evaluating Accuracy</a></li>
</ul></li>
<li class="chapter" data-level="24.3" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#wilcoxmann-whitney-u-test"><i class="fa fa-check"></i><b>24.3</b> Wilcox/Mann-Whitney-U Test</a></li>
<li class="chapter" data-level="24.4" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#edger"><i class="fa fa-check"></i><b>24.4</b> edgeR</a></li>
<li class="chapter" data-level="24.5" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#monocle-1"><i class="fa fa-check"></i><b>24.5</b> Monocle</a></li>
<li class="chapter" data-level="24.6" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#mast"><i class="fa fa-check"></i><b>24.6</b> MAST</a></li>
<li class="chapter" data-level="24.7" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#slow-methods-1h-to-run"><i class="fa fa-check"></i><b>24.7</b> Slow Methods (&gt;1h to run)</a></li>
<li class="chapter" data-level="24.8" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#bpsc"><i class="fa fa-check"></i><b>24.8</b> BPSC</a></li>
<li class="chapter" data-level="24.9" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#scde"><i class="fa fa-check"></i><b>24.9</b> SCDE</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="ideal-scrnaseq-pipeline-as-of-oct-2016.html"><a href="ideal-scrnaseq-pipeline-as-of-oct-2016.html"><i class="fa fa-check"></i><b>25</b> “Ideal” scRNAseq pipeline (as of Oct 2016)</a><ul>
<li class="chapter" data-level="25.1" data-path="ideal-scrnaseq-pipeline-as-of-oct-2016.html"><a href="ideal-scrnaseq-pipeline-as-of-oct-2016.html#experimental-design"><i class="fa fa-check"></i><b>25.1</b> Experimental Design</a></li>
<li class="chapter" data-level="25.2" data-path="ideal-scrnaseq-pipeline-as-of-oct-2016.html"><a href="ideal-scrnaseq-pipeline-as-of-oct-2016.html#processing-reads"><i class="fa fa-check"></i><b>25.2</b> Processing Reads</a></li>
<li class="chapter" data-level="25.3" data-path="ideal-scrnaseq-pipeline-as-of-oct-2016.html"><a href="ideal-scrnaseq-pipeline-as-of-oct-2016.html#preparing-expression-matrix"><i class="fa fa-check"></i><b>25.3</b> Preparing Expression Matrix</a></li>
<li class="chapter" data-level="25.4" data-path="ideal-scrnaseq-pipeline-as-of-oct-2016.html"><a href="ideal-scrnaseq-pipeline-as-of-oct-2016.html#biological-interpretation"><i class="fa fa-check"></i><b>25.4</b> Biological Interpretation</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="advanced-exercises.html"><a href="advanced-exercises.html"><i class="fa fa-check"></i><b>26</b> Advanced exercises</a></li>
<li class="chapter" data-level="27" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>27</b> Resources</a><ul>
<li class="chapter" data-level="27.1" data-path="resources.html"><a href="resources.html#scrna-seq-protocols"><i class="fa fa-check"></i><b>27.1</b> scRNA-seq protocols</a></li>
<li class="chapter" data-level="27.2" data-path="resources.html"><a href="resources.html#external-rna-control-consortium-ercc"><i class="fa fa-check"></i><b>27.2</b> External RNA Control Consortium (ERCC)</a></li>
<li class="chapter" data-level="27.3" data-path="resources.html"><a href="resources.html#scrna-seq-analysis-tools"><i class="fa fa-check"></i><b>27.3</b> scRNA-seq analysis tools</a></li>
</ul></li>
<li class="chapter" data-level="28" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>28</b> References</a></li>
<li class="divider"></li>
<li><a href="http://www.sanger.ac.uk/science/groups/hemberg-group" target="blank">Hemberg Lab, 2017</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Analysis of single cell RNA-seq data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="clustering-introduction" class="section level1">
<h1><span class="header-section-number">19</span> Clustering Introduction</h1>
<p>Once we have normalized the data and removed confounders we can carry out analyses that will allow us to interpret the data biologically. The exact nature of the analysis depends on the dataset and the biological question at hand. Nevertheless, there are a few operations which are useful in a wide range of contexts and we will be discussing some of them. We will start with the clustering of scRNA-seq data.</p>
<div id="introduction-7" class="section level2">
<h2><span class="header-section-number">19.1</span> Introduction</h2>
<p>One of the most promising applications of scRNA-seq is the discovery and annotation of cell-types based on the transcription profiles. Computationally, this is a hard problem as it amounts to <strong>unsupervised clustering</strong>. That is, we need to identify groups of cells based on the similarities of the transcriptomes without any prior knowledge of the labels. The problem is made more challenging due to the high level of noise and the large number of dimensions (i.e. genes).</p>
</div>
<div id="dimensionality-reductions" class="section level2">
<h2><span class="header-section-number">19.2</span> Dimensionality reductions</h2>
<p>When working with large datasets, it can often be beneficial to apply some sort of dimensionality reduction method. By projecting the data onto a lower-dimensional sub-space, one is often able to significantly reduce the amount of noise. An additional benefit is that it is typically much easier to visualize the data in a 2 or 3-dimensional subspace. Here we will introduce some of the popular dimensionality reduction methods.</p>
<div id="pca" class="section level3">
<h3><span class="header-section-number">19.2.1</span> PCA</h3>
<p>PCA analysis was introduced in chapter <a href="data-visualization.html#visual-pca">11.2</a>.</p>
</div>
<div id="spectral" class="section level3">
<h3><span class="header-section-number">19.2.2</span> Spectral</h3>
<p>Spectral decomposition is the factorization of a matrix into a canonical form, whereby the matrix is represented in terms of its eigenvalues and eigenvectors.</p>
<p>In application to scRNA-seq data, the matrix can be either an input expression matrix, or matrix of distances between the cells. The computed eigenvectors are similar to the projections of the data to PCs (chapter <a href="data-visualization.html#visual-pca">11.2</a>.).</p>
</div>
<div id="tsne" class="section level3">
<h3><span class="header-section-number">19.2.3</span> tSNE</h3>
<p>tSNE analysis was introduced in chapter <a href="data-visualization.html#visual-tsne">11.3</a>.</p>
</div>
</div>
<div id="clustering-methods" class="section level2">
<h2><span class="header-section-number">19.3</span> Clustering methods</h2>
<p><strong>Unsupervised clustering</strong> is useful in many different applications and it has been widely studied in machine learning. Some of the most popular approaches are discussed below.</p>
<div id="hierarchical-clustering" class="section level3">
<h3><span class="header-section-number">19.3.1</span> Hierarchical clustering</h3>
<p>In <a href="https://en.wikipedia.org/wiki/Hierarchical_clustering">hierarchical clustering</a>, one can use either a bottom-up or a top-down approach. In the former case, each cell is initially assigned to its own cluster and pairs of clusters are subsequently merged to create a hieararchy:</p>
<div class="figure" style="text-align: center"><span id="fig:clust-hierarch-raw"></span>
<img src="figures/hierarchical_clustering1.png" alt="Raw data" width="30%" />
<p class="caption">
Figure 19.1: Raw data
</p>
</div>
<div class="figure" style="text-align: center"><span id="fig:clust-hierarch-dendr"></span>
<img src="figures/hierarchical_clustering2.png" alt="The hierarchical clustering dendrogram" width="50%" />
<p class="caption">
Figure 19.2: The hierarchical clustering dendrogram
</p>
</div>
<p>With a top-down strategy, one instead starts with all observations in one cluster and then recursively split each cluster to form a hierarchy. One of the advantages of this strategy is that the method is deterministic.</p>
</div>
<div id="k-means" class="section level3">
<h3><span class="header-section-number">19.3.2</span> k-means</h3>
<p>In <a href="https://en.wikipedia.org/wiki/K-means_clustering"><em>k</em>-means clustering</a>, the goal is to partition <em>N</em> cells into <em>k</em> different clusters. In an iterative manner, cluster centers are assigned and each cell is assigned to its nearest cluster:</p>
<div class="figure" style="text-align: center"><span id="fig:clust-k-means"></span>
<img src="figures/k-means.png" alt="Schematic representation of the k-means clustering" width="100%" />
<p class="caption">
Figure 19.3: Schematic representation of the k-means clustering
</p>
</div>
<p>Most methods for scRNA-seq analysis includes a <em>k</em>-means step at some point.</p>
</div>
<div id="graph-based-methods" class="section level3">
<h3><span class="header-section-number">19.3.3</span> Graph-based methods</h3>
<p>Over the last two decades there has been a lot of interest in analyzing networks in various domains. One goal is to identify groups or modules of nodes in a network.</p>
<div class="figure" style="text-align: center"><span id="fig:clust-graph"></span>
<img src="figures/graph_network.jpg" alt="Schematic representation of the graph network" width="100%" />
<p class="caption">
Figure 19.4: Schematic representation of the graph network
</p>
</div>
<p>Some of these methods can be applied to scRNA-seq data and one example is the method, which is based on the concept of identifying groups of tightly connected nodes.</p>
</div>
</div>
<div id="challenges-in-clustering" class="section level2">
<h2><span class="header-section-number">19.4</span> Challenges in clustering</h2>
<ul>
<li>What is the number of clusters <em>k</em>?</li>
<li><strong>Scalability</strong>: in the last 2 years the number of cells in scRNA-seq experiments has grown by 2 orders of magnitude from ~<span class="math inline">\(10^2\)</span> to ~<span class="math inline">\(10^4\)</span></li>
<li>Tools are not user-friendly</li>
</ul>
</div>
<div id="tools-for-scrna-seq-data" class="section level2">
<h2><span class="header-section-number">19.5</span> Tools for scRNA-seq data</h2>
<div id="sincera" class="section level3">
<h3><span class="header-section-number">19.5.1</span> <a href="https://research.cchmc.org/pbge/sincera.html">SINCERA</a></h3>
<ul>
<li>SINCERA <span class="citation">(Guo et al. <a href="#ref-Guo2015-ok">2015</a>)</span> is based on hierarchical clustering</li>
<li>Data is converted to <em>z</em>-scores before clustering</li>
<li>Identify <em>k</em> by finding the first singleton cluster in the hierarchy</li>
</ul>
</div>
<div id="pcareduce" class="section level3">
<h3><span class="header-section-number">19.5.2</span> <a href="https://github.com/JustinaZ/pcaReduce">pcaReduce</a></h3>
<p>pcaReduce <span class="citation">(Žurauskienė and Yau <a href="#ref-Zurauskiene2016-kg">2016</a>)</span> combines PCA, <em>k</em>-means and “iterative” hierarchical clustering. Starting from a large number of clusters pcaReduce iteratively merges similar clusters; after each merging event it removes the principle component explaning the least variance in the data.</p>
</div>
<div id="sc3" class="section level3">
<h3><span class="header-section-number">19.5.3</span> <a href="http://bioconductor.org/packages/SC3/">SC3</a></h3>
<div class="figure" style="text-align: center"><span id="fig:clust-sc3"></span>
<img src="figures/sc3.png" alt="SC3 pipeline" width="100%" />
<p class="caption">
Figure 19.5: SC3 pipeline
</p>
</div>
<ul>
<li>SC3 <span class="citation">(Kiselev et al. <a href="#ref-Kiselev2016-bq">2016</a>)</span> is based on PCA and spectral dimensionality reductions</li>
<li>Utilises <em>k</em>-means</li>
<li>Additionally performs the consensus clustering</li>
</ul>
</div>
<div id="tsne-k-means" class="section level3">
<h3><span class="header-section-number">19.5.4</span> tSNE + k-means</h3>
<ul>
<li>Based on tSNE maps</li>
<li>Utilises <em>k</em>-means</li>
</ul>
</div>
<div id="seurat" class="section level3">
<h3><span class="header-section-number">19.5.5</span> <a href="https://github.com/Puriney/seurat">SEURAT</a></h3>
<p>SEURAT <span class="citation">(Macosko et al. <a href="#ref-Macosko2015-ix">2015</a>)</span> first utilises PCA on a set of cells, then a number of statistically significant PCs is defined. Those PCs are further projected to a 2D space using tSNE. The remaining cells are then projected on the same tSNE map. Density clustering algorithm (<a href="https://en.wikipedia.org/wiki/DBSCAN">DBSCAN</a>) is then used to identify cell clusters in the 2D space.</p>
<p><strong>Note</strong> In the newest versions of SEURAT (v. 1.3-1.4) the tSNE is now used exclusively for visualization, and clustering is based on a <em>community detection</em> approach similar to one previously proposed for analyzing CyTOF data <span class="citation">(Levine et al. <a href="#ref-Levine2015-fk">2015</a>)</span>.</p>
</div>
<div id="snn-cliq" class="section level3">
<h3><span class="header-section-number">19.5.6</span> <a href="http://bioinfo.uncc.edu/SNNCliq/">SNN-Cliq</a></h3>
<p>SNN-Cliq <span class="citation">(C. Xu and Su <a href="#ref-Xu2015-vf">2015</a>)</span> is a graph-based method. First the method identifies the k-nearest-neighbours of each cell according to the <em>distance</em> measure. This is used to calculate the number of Shared Nearest Neighbours (SNN) between each pair of cells. A graph is built by placing an edge between two cells If they have at least one SNN. Clusters are defined as groups of cells with many edges between them using a “clique” method. SNN-Cliq requires several parameters to be defined manually.</p>

</div>
</div>
</div>
<h3> References</h3>
<div id="refs" class="references">
<div id="ref-Guo2015-ok">
<p>Guo, Minzhe, Hui Wang, S Steven Potter, Jeffrey A Whitsett, and Yan Xu. 2015. “SINCERA: A Pipeline for Single-Cell RNA-Seq Profiling Analysis.” <em>PLoS Comput. Biol.</em> 11 (11): e1004575.</p>
</div>
<div id="ref-Zurauskiene2016-kg">
<p>Žurauskienė, Justina, and Christopher Yau. 2016. “PcaReduce: Hierarchical Clustering of Single Cell Transcriptional Profiles.” <em>BMC Bioinformatics</em> 17 (22~mar): 140.</p>
</div>
<div id="ref-Kiselev2016-bq">
<p>Kiselev, Vladimir Yu, Kristina Kirschner, Michael T Schaub, Tallulah Andrews, Tamir Chandra, Kedar N Natarajan, Wolf Reik, Mauricio Barahona, Anthony R Green, and Martin Hemberg. 2016. “SC3 - Consensus Clustering of Single-Cell RNA-Seq Data.” <em>BioRxiv</em>.</p>
</div>
<div id="ref-Macosko2015-ix">
<p>Macosko, Evan Z, Anindita Basu, Rahul Satija, James Nemesh, Karthik Shekhar, Melissa Goldman, Itay Tirosh, et al. 2015. “Highly Parallel Genome-Wide Expression Profiling of Individual Cells Using Nanoliter Droplets.” <em>Cell</em> 161 (5): 1202–14.</p>
</div>
<div id="ref-Levine2015-fk">
<p>Levine, Jacob H, Erin F Simonds, Sean C Bendall, Kara L Davis, El-Ad D Amir, Michelle D Tadmor, Oren Litvin, et al. 2015. “Data-Driven Phenotypic Dissection of AML Reveals Progenitor-Like Cells That Correlate with Prognosis.” <em>Cell</em> 162 (1): 184–97.</p>
</div>
<div id="ref-Xu2015-vf">
<p>Xu, Chen, and Zhengchang Su. 2015. “Identification of Cell Types from Single-Cell Transcriptomes Using a Novel Clustering Method.” <em>Bioinformatics</em>, 11~feb.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="dealing-with-confounders-reads.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="clust-methods.html" class="navigation navigation-next " aria-label="Next page""><i class="fa fa-angle-right"></i></a>

<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/hemberg-lab/scRNA.seq.course/edit/master/17-clust-intro.Rmd",
"text": "Edit"
},
"download": ["scRNA-seq-course.pdf"],
"toc": {
"collapse": "section"
},
"search": true
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
